{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3434122\n",
      "torch.Size([196608, 4])\n"
     ]
    }
   ],
   "source": [
    "import torch\n",
    "import polars as pl\n",
    "from enformer_pytorch import Enformer, GenomeIntervalDataset\n",
    "\n",
    "# filter_train = lambda df: df.filter(pl.col('column_4') == 'train')\n",
    "\n",
    "ds = GenomeIntervalDataset(\n",
    "    bed_file = '/home/hxcai/cell_type_specific_CRE/data/ATAC_seq/auged_sorted_GM12878_CTCF_intersect.bed', # bed file - columns 0, 1, 2 must be <chromosome>, <start position>, <end position>\n",
    "    fasta_file = '/home/hxcai/ref_genome/hg38.fa',         # path to fasta file\n",
    "    # filter_df_fn = filter_train,                        # filter dataframe function\n",
    "    return_seq_indices = False,                          # return nucleotide indices (ACGTN) or one hot encodings\n",
    "    shift_augs = (-2, 2),                               # random shift augmentations from -2 to +2 basepairs\n",
    "    context_length = 196_608,\n",
    ")\n",
    "\n",
    "print(len(ds))\n",
    "print(ds[0].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1, 1000, 4)\n",
      "(1, 1)\n"
     ]
    }
   ],
   "source": [
    "import selene_sdk\n",
    "from selene_sdk import samplers, sequences, utils\n",
    "\n",
    "# reference_sequence = sequences.Genome(input_path = '/home/hxcai/ref_genome/hg38.fa', blacklist_regions='hg38')\n",
    "# features = utils.load_features_list('/home/hxcai/cell_type_specific_CRE/data/ATAC_seq/distinct_features.txt')\n",
    "\n",
    "sampler = samplers.IntervalsSampler(\n",
    "    reference_sequence = sequences.Genome(input_path = '/home/hxcai/ref_genome/hg38.fa', blacklist_regions='hg38'), \n",
    "    features = utils.load_features_list('/home/hxcai/cell_type_specific_CRE/data/ATAC_seq/distinct_features.txt'),\n",
    "    target_path = '/home/hxcai/cell_type_specific_CRE/data/ATAC_seq/HepG2_ENCFF913MQB.bed',\n",
    "    intervals_path = '/home/hxcai/cell_type_specific_CRE/data/ATAC_seq/deepsea_TF_intervals.txt',\n",
    "    sample_negative = True,\n",
    "    seed = 436,\n",
    "    validation_holdout = ['chr6', 'chr7'],\n",
    "    test_holdout = ['chr8', 'chr9'],\n",
    "    sequence_length = 1000, \n",
    "    center_bin_to_predict = 200, \n",
    "    feature_thresholds = 0.5, \n",
    "    mode = \"train\", \n",
    "    save_datasets = [\"test\"], \n",
    "    output_dir = None)\n",
    "\n",
    "x = sampler.sample(batch_size=1)\n",
    "# print(len(sampler))\n",
    "print(x[0].shape)\n",
    "print(x[1].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3434122\n",
      "(tensor([[0.2500, 0.2500, 0.2500,  ..., 0.0000, 0.0000, 0.0000],\n",
      "        [0.2500, 0.2500, 0.2500,  ..., 0.0000, 1.0000, 0.0000],\n",
      "        [0.2500, 0.2500, 0.2500,  ..., 1.0000, 0.0000, 0.0000],\n",
      "        [0.2500, 0.2500, 0.2500,  ..., 0.0000, 0.0000, 1.0000]]), tensor([0.]))\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "sys.path.append('/home/hxcai/cell_type_specific_CRE')\n",
    "from datasets import BedDataset\n",
    "\n",
    "d = BedDataset(bed_file= '/home/hxcai/cell_type_specific_CRE/data/ATAC_seq/auged_sorted_GM12878_CTCF_intersect.bed',\n",
    "        ref_genome_path= '/home/hxcai/ref_genome/hg19.fa',\n",
    "        context_length= 1000,\n",
    "        filter_in_list=['chr1']\n",
    "        )\n",
    "\n",
    "print(len(d))\n",
    "print(d[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3434122\n",
      "{'inputs': array([[0.25, 0.25, 0.25, 0.25],\n",
      "       [0.25, 0.25, 0.25, 0.25],\n",
      "       [0.25, 0.25, 0.25, 0.25],\n",
      "       ...,\n",
      "       [0.  , 0.  , 1.  , 0.  ],\n",
      "       [0.  , 1.  , 0.  , 0.  ],\n",
      "       [0.  , 0.  , 0.  , 1.  ]]), 'targets': array([0.]), 'metadata': {'ranges': GenomicRanges(chr='chr1', start=9600, end=10600, id='0', strand='*')}}\n"
     ]
    }
   ],
   "source": [
    "from kipoiseq.dataloaders import SeqIntervalDl\n",
    "d = SeqIntervalDl(\n",
    "    intervals_file='/home/hxcai/cell_type_specific_CRE/data/ATAC_seq/auged_sorted_GM12878_CTCF_intersect.bed',\n",
    "    fasta_file='/home/hxcai/ref_genome/hg19.fa',\n",
    "    auto_resize_len=1000,\n",
    "    \n",
    ")\n",
    "print(len(d))\n",
    "print(d[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import selene_sdk\n",
    "from selene_sdk import samplers, sequences, utils\n",
    "\n",
    "reference_sequence = sequences.Genome(input_path = '/home/hxcai/ref_genome/hg38.fa', blacklist_regions='hg38')\n",
    "features = utils.load_features_list('/home/hxcai/cell_type_specific_CRE/data/ATAC_seq/distinct_features.txt')\n",
    "\n",
    "sampler = samplers.IntervalsSampler(\n",
    "    reference_sequence = reference_sequence, \n",
    "    features = features,\n",
    "    target_path = '/home/hxcai/cell_type_specific_CRE/data/ATAC_seq/HepG2_ENCFF913MQB.bed',\n",
    "    intervals_path = '/home/hxcai/cell_type_specific_CRE/data/ATAC_seq/deepsea_TF_intervals.txt',\n",
    "    sample_negative = True,\n",
    "    seed = 436,\n",
    "    validation_holdout = ['chr6', 'chr7'],\n",
    "    test_holdout = ['chr8', 'chr9'],\n",
    "    sequence_length = 1000, \n",
    "    center_bin_to_predict = 200, \n",
    "    feature_thresholds = 0.5, \n",
    "    mode = \"train\", \n",
    "    save_datasets = [\"test\"], \n",
    "    output_dir = None)\n",
    "\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    x = sampler.sample(batch_size=1)\n",
    "    print(x[0].shape)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "torch",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "vscode": {
   "interpreter": {
    "hash": "8e62da015c2b8e1460a389c60847c46fcac689812463b880d1463e5f1a16b0d6"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
