{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os\n",
    "os.chdir('../../data/Gosai_MPRA')\n",
    "\n",
    "import sys\n",
    "sys.path.append('/home/hxcai/cell_type_specific_CRE')\n",
    "\n",
    "from utils import *\n",
    "# from datasets import GenomeInterval\n",
    "from pyfaidx import Fasta\n",
    "\n",
    "from Bio import pairwise2\n",
    "from Bio.pairwise2 import format_alignment, align\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome = Fasta('/home/hxcai/genome/hg19.fa')\n",
    "seq3 = genome['chr15'][89574340: 89574540].seq\n",
    "seq3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rc_seq(seq3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq1 = 'CATAAAGGCAGTGTAGACCCAAACAGTGAGCAGTAGCAAGATTTATTACAAAGAGCGAAAGAAGAACGAAACCACATCGCAAAACGGAACTCCAGCCGGTTGCCACTACTGCCTCGGGCAGCCTGCTTTTATTCTCTTATCTGGCCCCACCCACATCCTGCTGATTGGTCCATTTTACAGAGAGTGGATTGGTCCATTT'\n",
    "seq2 = 'CATAAAGGCAGTGTAGACCCAAACAGTGAGCAGTAGCAAGATTTATTACAAAGAGCGAAAGAAGAACGAAACCACATCGCAAAACGGAACTCCAGCCGGTTTGCCACTACTGCCTCGGGCAGCCTGCTTTTATTCTCTTATCTGGCCCCACCCACATCCTGCTGATTGGTCCATTTTACAGAGAGTGGATTGGTCCATTT'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "alignments = align.globalms(seq1, seq2, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "\n",
    "alignments = align.globalms(seq1, rc_seq(seq3), 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 12:63513920:G:A:A:wC  G->A\n",
    "# 12:63513920:G:A:R:wC  G->G\n",
    "\n",
    "# 2:145626403:A:C:R:wC  A->A\n",
    "# 2:145626403:A:C:A:wC  A->C\n",
    "\n",
    "# 第三个是A的话,替换\n",
    "# 第三个是R的话,不变\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "print(genome['chr12'][63513920])\n",
    "print(genome['chr12'][63513915:63513925])\n",
    "\n",
    "seq1 = 'CATAAAGGGCTGAACATGCTGTTGAAAAAATGTAGATATAAAAGTTCATAAGAAATATTGATTCAATTTTTATGACTATGTAATTCTCATTTTTTTCAGATGCCATGTCAAAATATATTTGTTCCTCTTCTAGTTTTAGGATAATGAGACTGCTATGTGATATAACTACTTGGATGAATATTATCAGAAATCTAGATTTA'\n",
    "seq2 = 'CATAAAGGGCTGAACATGCTGTTGAAAAAATGTAGATATAAAAGTTCATAAGAAATATTGATTCAATTTTTATGACTATGTAATTCTCATTTTTTTCAGGTGCCATGTCAAAATATATTTGTTCCTCTTCTAGTTTTAGGATAATGAGACTGCTATGTGATATAACTACTTGGATGAATATTATCAGAAATCTAGATTTA'\n",
    "print(len(seq1), len(seq2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq3 = genome['chr12'][63513820:63514020].seq\n",
    "# seq3 = rc_seq(seq3)\n",
    "alignments = align.globalms(seq1, seq2, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "alignments = align.globalms(seq1, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "alignments = align.globalms(seq2, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2:145626403:A:C:R:wC  A->A\n",
    "# 2:145626403:A:C:A:wC  A->C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq1 = 'CATAAAGGTAGACCATATGGTCTAGGAAGCAGAGGACAGCTCAGAAACAGAGAAGAACTGAACATATGTGCAAAGAAGAGAGGAGAGATGGGACAGAGAAGACAGACCTCATCTTTATAAGTTCTTGATTGCATCTTTTCCCTGAGACCTGGCCACATCCCTAGCCCTAGACTTCATGAGACAGCCCAGTATCTTTAAAA'\n",
    "seq2 = 'CATAAAGGTAGACCATATGGTCTAGGAAGCAGAGGACAGCTCAGAAACAGAGAAGAACTGAACATATGTGCAAAGAAGAGAGGAGAGATGGGACAGAGACGACAGACCTCATCTTTATAAGTTCTTGATTGCATCTTTTCCCTGAGACCTGGCCACATCCCTAGCCCTAGACTTCATGAGACAGCCCAGTATCTTTAAAA'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq3 = genome['chr2'][145626303: 145626503].seq\n",
    "seq3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "alignments = align.globalms(seq1, seq2, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "alignments = align.globalms(seq1, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "alignments = align.globalms(seq2, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 7:4682252:T:TTTG:A:wC\n",
    "# 7:4682252:T:TTTG:R:wC\n",
    "\n",
    "seq1 = 'CATAAATAAATAAACGAATGGGGAGGAAATTGATAAAATATTAAAAGAGTGTTGTTTCAGAGTTTTTACTATGTCACCTCTTACAGATTTATGAGAGTTTGTTGTTGTTTTTTGGAGAAGAGGGGTCTCCCTATGTTACCCAGGCTGGTCTCAAACTCCTGGGCTCAATCCATCCGTCCGCCTCAGCCTCCTGAAGTGCT'\n",
    "seq2 = 'CATAAATAAATAAACGAATGGGGAGGAAATTGATAAAATATTAAAAGAGTGTTGTTTCAGAGTTTTTACTATGTCACCTCTTACAGATTTATGAGAGTTTGTTGTTTTTTGGAGAAGAGGGGTCTCCCTATGTTACCCAGGCTGGTCTCAAACTCCTGGGCTCAATCCATCCGTCCGCCTCAGCCTCCTGAAGTGCT'\n",
    "seq3 = genome['chr7'][4682152: 4682352].seq.upper()\n",
    "print(len(seq1), len(seq2), len(seq3))\n",
    "\n",
    "alignments = align.globalms(seq1, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "alignments = align.globalms(seq2, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('supp_materials/SupTable 2 - UKBB_GTEX_CODA_averaged_no_cutoffs.txt', delimiter='\\t', low_memory=False)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['hg19_pos'] = df['IDs'].str.split(':').str[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "match_indices = []\n",
    "genome = Fasta('/home/hxcai/genome/hg19.fa')\n",
    "\n",
    "for i, row in tqdm(df.iterrows(), total=len(df)):\n",
    "    IDs = row['IDs'].split(':')\n",
    "    if len(IDs) >=5:\n",
    "        if IDs[4] == 'R':\n",
    "            # R means reference 和基因组相同\n",
    "            if len(IDs[2]) == 1 and len(IDs[3]) == 1:\n",
    "                # 单碱基替换\n",
    "                chr = 'chr'+IDs[0]\n",
    "                pos = int(IDs[1])\n",
    "                genome_seq = genome[chr][pos-100:pos+100].seq.upper()\n",
    "                data_seq = row['nt_sequence'].upper()\n",
    "                if genome_seq == data_seq:\n",
    "                    match_indices.append(i)\n",
    "                else:\n",
    "                    print(f'{i}, not match')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(match_indices)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.loc[match_indices, ['nt_sequence', 'chr', 'hg19_pos', 'HepG2_mean', 'K562_mean', 'SKNSH_mean']]\n",
    "df.columns = ['seq', 'chr', 'hg19_pos', 'K562', 'HepG2', 'SKNSH']\n",
    "df['chr'] = 'chr' + df['chr'].astype(str)\n",
    "df.to_csv('Gosai_MPRA_match_genome.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pyliftover import LiftOver\n",
    "\n",
    "# 创建LiftOver对象，hg19到hg38\n",
    "lo = LiftOver('hg19', 'hg38')\n",
    "\n",
    "df = pd.read_csv('Gosai_MPRA_match_genome.csv', sep='\\t')\n",
    "\n",
    "df['hg38_pos'] = np.nan\n",
    "\n",
    "for i, row in tqdm(df.iterrows(), total=len(df)):\n",
    "    chr, hg19_pos = row['chr'], row['hg19_pos']\n",
    "    converted = lo.convert_coordinate(chr, hg19_pos)\n",
    "    # 检查转换结果是否为空\n",
    "    if converted and len(converted) > 0:\n",
    "        hg38_pos = converted[0][1]  # 成功转换的新位置\n",
    "        df.at[i, 'hg38_pos'] = hg38_pos\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "df = df.dropna().reset_index(drop=True)\n",
    "df['hg38_pos'] = df['hg38_pos'].astype(int)\n",
    "df = df.sort_values(by=['chr', 'hg38_pos'])\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.dropna().reset_index(drop=True)\n",
    "df['hg38_pos'] = df['hg38_pos'].astype(int)\n",
    "df = df.sort_values(by=['chr', 'hg38_pos'])\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('Gosai_MPRA_match_genome.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "subprocess.run(\"bedtools intersect -a genome_pos.bed -b ../ATAC/HepG2_unique.bed -wa -f 0.5 > genome_pos_intersect_HepG2.bed\", shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersect = pd.read_csv('genome_pos_intersect_HepG2.bed', sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "match_indices\n",
    "HepG2_pos_indices = intersect[3]\n",
    "HepG2_neg_indices = [i for i in match_indices if i not in HepG2_pos_indices]\n",
    "print(len(HepG2_pos_indices), len(HepG2_neg_indices))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "HepG2_pos_MPRA_exp = df['HepG2_mean'][HepG2_pos_indices]\n",
    "HepG2_neg_MPRA_exp = df['HepG2_mean'][HepG2_neg_indices]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.histplot(HepG2_pos_MPRA_exp, stat='density', color='orange', alpha=0.5)\n",
    "sns.histplot(HepG2_neg_MPRA_exp, stat='density', color='blue',   alpha=0.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "torch",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
