{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hxcai/anaconda3/envs/torch/lib/python3.10/site-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "from Bio import pairwise2\n",
    "from Bio.pairwise2 import format_alignment, align\n",
    "\n",
    "import sys\n",
    "sys.path.append('..')\n",
    "from MPRA_exp.utils import *\n",
    "os.chdir('../../data/SirajMPRA')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df4 = pd.read_excel('supp/media-4.xlsx')\n",
    "df4\n",
    "\n",
    "# # a = np.log2((df['mean_RNA_ref_HEPG2'] / df['mean_Plasmid_ref_HEPG2']).astype(float))\n",
    "# # b = df['allele1.log2FC_HEPG2']\n",
    "# # sns.scatterplot(x=a[:10000], y=b[:10000], s = 1)\n",
    "\n",
    "# select_columns = ['allele1.log2FC_A549', 'allele1.log2FC_HEPG2', 'allele1.log2FC_K562', 'allele1.log2FC_SKNSH', 'allele1.log2FC_HCT116', \n",
    "#                   'allele2.log2FC_A549', 'allele2.log2FC_HEPG2', 'allele2.log2FC_K562', 'allele2.log2FC_SKNSH', 'allele2.log2FC_HCT116']\n",
    "# df = df[['variant'] + select_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df4.columns = df4.iloc[0]\n",
    "df4 = df4.iloc[1:].reset_index(drop=True)\n",
    "df4 = df4.rename(columns={'variant': 'Variant'})\n",
    "df4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Supplementary Table 2 |Variant Oligo Sequences</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Variant</td>\n",
       "      <td>Allele 1 Oligo</td>\n",
       "      <td>Allele 2 Oligo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:522224:G:A</td>\n",
       "      <td>TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...</td>\n",
       "      <td>TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:729679:C:G</td>\n",
       "      <td>TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...</td>\n",
       "      <td>TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:741397:A:G</td>\n",
       "      <td>TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...</td>\n",
       "      <td>TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365758</th>\n",
       "      <td>chr4:57975328:A:G</td>\n",
       "      <td>TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...</td>\n",
       "      <td>TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365759</th>\n",
       "      <td>chr20:36661369:C:A</td>\n",
       "      <td>GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...</td>\n",
       "      <td>GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365760</th>\n",
       "      <td>chr8:60683160:C:T</td>\n",
       "      <td>GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...</td>\n",
       "      <td>GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365761</th>\n",
       "      <td>chr7:31046388:G:C</td>\n",
       "      <td>CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...</td>\n",
       "      <td>CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365762</th>\n",
       "      <td>chr10:10216373:A:G</td>\n",
       "      <td>GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...</td>\n",
       "      <td>GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>365763 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Supplementary Table 2 |Variant Oligo Sequences  \\\n",
       "0                                             Variant   \n",
       "1                                      chr1:14677:G:A   \n",
       "2                                     chr1:522224:G:A   \n",
       "3                                     chr1:729679:C:G   \n",
       "4                                     chr1:741397:A:G   \n",
       "...                                               ...   \n",
       "365758                              chr4:57975328:A:G   \n",
       "365759                             chr20:36661369:C:A   \n",
       "365760                              chr8:60683160:C:T   \n",
       "365761                              chr7:31046388:G:C   \n",
       "365762                             chr10:10216373:A:G   \n",
       "\n",
       "                                               Unnamed: 1  \\\n",
       "0                                          Allele 1 Oligo   \n",
       "1       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "2       TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...   \n",
       "3       TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...   \n",
       "4       TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...   \n",
       "...                                                   ...   \n",
       "365758  TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...   \n",
       "365759  GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...   \n",
       "365760  GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...   \n",
       "365761  CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...   \n",
       "365762  GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...   \n",
       "\n",
       "                                               Unnamed: 2  \n",
       "0                                          Allele 2 Oligo  \n",
       "1       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...  \n",
       "2       TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...  \n",
       "3       TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...  \n",
       "4       TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...  \n",
       "...                                                   ...  \n",
       "365758  TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...  \n",
       "365759  GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...  \n",
       "365760  GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...  \n",
       "365761  CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...  \n",
       "365762  GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...  \n",
       "\n",
       "[365763 rows x 3 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3 = pd.read_excel('supp/media-3.xlsx')\n",
    "df3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Allele 1 Oligo</th>\n",
       "      <th>Allele 2 Oligo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:522224:G:A</td>\n",
       "      <td>TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...</td>\n",
       "      <td>TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:729679:C:G</td>\n",
       "      <td>TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...</td>\n",
       "      <td>TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:741397:A:G</td>\n",
       "      <td>TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...</td>\n",
       "      <td>TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1:752566:G:A</td>\n",
       "      <td>AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...</td>\n",
       "      <td>AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365758</th>\n",
       "      <td>chr4:57975328:A:G</td>\n",
       "      <td>TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...</td>\n",
       "      <td>TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365759</th>\n",
       "      <td>chr20:36661369:C:A</td>\n",
       "      <td>GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...</td>\n",
       "      <td>GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365760</th>\n",
       "      <td>chr8:60683160:C:T</td>\n",
       "      <td>GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...</td>\n",
       "      <td>GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365761</th>\n",
       "      <td>chr7:31046388:G:C</td>\n",
       "      <td>CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...</td>\n",
       "      <td>CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365762</th>\n",
       "      <td>chr10:10216373:A:G</td>\n",
       "      <td>GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...</td>\n",
       "      <td>GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>365762 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0                  Variant                                     Allele 1 Oligo  \\\n",
       "1           chr1:14677:G:A  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "2          chr1:522224:G:A  TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...   \n",
       "3          chr1:729679:C:G  TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...   \n",
       "4          chr1:741397:A:G  TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...   \n",
       "5          chr1:752566:G:A  AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...   \n",
       "...                    ...                                                ...   \n",
       "365758   chr4:57975328:A:G  TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...   \n",
       "365759  chr20:36661369:C:A  GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...   \n",
       "365760   chr8:60683160:C:T  GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...   \n",
       "365761   chr7:31046388:G:C  CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...   \n",
       "365762  chr10:10216373:A:G  GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...   \n",
       "\n",
       "0                                          Allele 2 Oligo  \n",
       "1       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...  \n",
       "2       TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...  \n",
       "3       TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...  \n",
       "4       TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...  \n",
       "5       AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...  \n",
       "...                                                   ...  \n",
       "365758  TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...  \n",
       "365759  GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...  \n",
       "365760  GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...  \n",
       "365761  CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...  \n",
       "365762  GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...  \n",
       "\n",
       "[365762 rows x 3 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3.columns = df3.iloc[0]\n",
    "df3 = df3.iloc[1:]\n",
    "df3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>active.any</th>\n",
       "      <th>emVar.any</th>\n",
       "      <th>allele1.log2FC_A549</th>\n",
       "      <th>allele1.log2FC_HEPG2</th>\n",
       "      <th>allele1.log2FC_K562</th>\n",
       "      <th>allele1.log2FC_SKNSH</th>\n",
       "      <th>allele1.log2FC_HCT116</th>\n",
       "      <th>allele1.log2FC.SE_A549</th>\n",
       "      <th>allele1.log2FC.SE_HEPG2</th>\n",
       "      <th>...</th>\n",
       "      <th>mean_RNA_ref_K562</th>\n",
       "      <th>mean_RNA_ref_SKNSH</th>\n",
       "      <th>mean_RNA_ref_HCT116</th>\n",
       "      <th>mean_RNA_alt_A549</th>\n",
       "      <th>mean_RNA_alt_HEPG2</th>\n",
       "      <th>mean_RNA_alt_K562</th>\n",
       "      <th>mean_RNA_alt_SKNSH</th>\n",
       "      <th>mean_RNA_alt_HCT116</th>\n",
       "      <th>Allele 1 Oligo</th>\n",
       "      <th>Allele 2 Oligo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100025816:G:A</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.623685</td>\n",
       "      <td>2.044789</td>\n",
       "      <td>1.22237</td>\n",
       "      <td>1.280575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155521</td>\n",
       "      <td>...</td>\n",
       "      <td>593.166667</td>\n",
       "      <td>305.4</td>\n",
       "      <td>459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>719.6</td>\n",
       "      <td>796</td>\n",
       "      <td>599.4</td>\n",
       "      <td>656.2</td>\n",
       "      <td>TCTGACCCCTCCCACGCTTCCTTTCCAGACTGAAACACTGTCTTCT...</td>\n",
       "      <td>TCTGACCCCTCCCACGCTTCCTTTCCAGACTGAAACACTGTCTTCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100029561:C:T</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.390524</td>\n",
       "      <td>0.296395</td>\n",
       "      <td>3.402503</td>\n",
       "      <td>2.162766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.088808</td>\n",
       "      <td>...</td>\n",
       "      <td>665.166667</td>\n",
       "      <td>5250</td>\n",
       "      <td>3554.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2298.8</td>\n",
       "      <td>439.333333</td>\n",
       "      <td>1873.8</td>\n",
       "      <td>1458.4</td>\n",
       "      <td>GCAACACAGAGGGTTCGGAAACGTGGGCTCTGCCTGCTGCCAATCA...</td>\n",
       "      <td>GCAACACAGAGGGTTCGGAAACGTGGGCTCTGCCTGCTGCCAATCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100098716:G:T</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.174309</td>\n",
       "      <td>0.853279</td>\n",
       "      <td>1.283911</td>\n",
       "      <td>1.000218</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.071422</td>\n",
       "      <td>...</td>\n",
       "      <td>1131.666667</td>\n",
       "      <td>2423</td>\n",
       "      <td>2850</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1243.4</td>\n",
       "      <td>374.333333</td>\n",
       "      <td>1114.2</td>\n",
       "      <td>1060.4</td>\n",
       "      <td>CCCCCAGCAAACACGGATCTGCTCTCTGTCACTATCGATTAGCTTG...</td>\n",
       "      <td>CCCCCAGCAAACACGGATCTGCTCTCTGTCACTATCGATTAGCTTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100109592:T:C</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.007378</td>\n",
       "      <td>0.301644</td>\n",
       "      <td>0.592409</td>\n",
       "      <td>-0.03329</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.139805</td>\n",
       "      <td>...</td>\n",
       "      <td>601.666667</td>\n",
       "      <td>643.4</td>\n",
       "      <td>718.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>695.2</td>\n",
       "      <td>522.5</td>\n",
       "      <td>802.6</td>\n",
       "      <td>754</td>\n",
       "      <td>GATCTCAGGCATCCTCACTTCGTGCCGAGAGGAGGCTCTTAAACAT...</td>\n",
       "      <td>GATCTCAGGCATCCTCACTTCGTGCCGAGAGGAGGCTCTTAAACAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100141027:T:C</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.238128</td>\n",
       "      <td>-0.00355</td>\n",
       "      <td>-0.288184</td>\n",
       "      <td>0.041701</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.076976</td>\n",
       "      <td>...</td>\n",
       "      <td>361.666667</td>\n",
       "      <td>417</td>\n",
       "      <td>731.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>627.8</td>\n",
       "      <td>374.666667</td>\n",
       "      <td>333.2</td>\n",
       "      <td>660.4</td>\n",
       "      <td>CTCCTTTACAGGGAACATTTGCCTGGAGAACAGCTGCACCCTTGTG...</td>\n",
       "      <td>CTCCTTTACAGGGAACATTTGCCTGGAGAACAGCTGCACCCTTGTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304273</th>\n",
       "      <td>chrX:99995091:A:G</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.171016</td>\n",
       "      <td>0.41844</td>\n",
       "      <td>-0.541239</td>\n",
       "      <td>0.296166</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.136126</td>\n",
       "      <td>...</td>\n",
       "      <td>390.166667</td>\n",
       "      <td>281</td>\n",
       "      <td>794</td>\n",
       "      <td>NaN</td>\n",
       "      <td>211.6</td>\n",
       "      <td>226</td>\n",
       "      <td>140.2</td>\n",
       "      <td>320.4</td>\n",
       "      <td>TAATTTATTCAGGAAAAATCTGAGGTCACTAACATTTACTAGTGCT...</td>\n",
       "      <td>TAATTTATTCAGGAAAAATCTGAGGTCACTAACATTTACTAGTGCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304274</th>\n",
       "      <td>chrX:99995933:C:G</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.433557</td>\n",
       "      <td>0.23761</td>\n",
       "      <td>0.368939</td>\n",
       "      <td>-0.133435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.064148</td>\n",
       "      <td>...</td>\n",
       "      <td>562.666667</td>\n",
       "      <td>884.2</td>\n",
       "      <td>948.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1853.8</td>\n",
       "      <td>657.833333</td>\n",
       "      <td>1382.6</td>\n",
       "      <td>1364.2</td>\n",
       "      <td>AGAGAGACTGACATCTCACTCCCAGGCTGGAGGCTTGCTGCACTAC...</td>\n",
       "      <td>AGAGAGACTGACATCTCACTCCCAGGCTGGAGGCTTGCTGCACTAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304275</th>\n",
       "      <td>chrX:99996102:A:G</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.09405</td>\n",
       "      <td>0.310849</td>\n",
       "      <td>0.070434</td>\n",
       "      <td>0.048445</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.060897</td>\n",
       "      <td>...</td>\n",
       "      <td>953.333333</td>\n",
       "      <td>1182</td>\n",
       "      <td>1679.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1936</td>\n",
       "      <td>908.166667</td>\n",
       "      <td>1137.8</td>\n",
       "      <td>1626.6</td>\n",
       "      <td>AAGCCTGGTTTAAATTGCTCATAATGGCAGAGCAAAGAAAGAGATC...</td>\n",
       "      <td>AAGCCTGGTTTAAATTGCTCATAATGGCAGAGCAAAGAAAGAGATC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304276</th>\n",
       "      <td>chrX:99998829:C:T</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.496841</td>\n",
       "      <td>0.125072</td>\n",
       "      <td>0.011764</td>\n",
       "      <td>-0.355709</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.132722</td>\n",
       "      <td>...</td>\n",
       "      <td>123.666667</td>\n",
       "      <td>207.4</td>\n",
       "      <td>265</td>\n",
       "      <td>NaN</td>\n",
       "      <td>306.4</td>\n",
       "      <td>118.166667</td>\n",
       "      <td>206.8</td>\n",
       "      <td>402.6</td>\n",
       "      <td>TTATTATCTCCACTTTACAGATGAGAAAACTGTGGCTCAGATAAGT...</td>\n",
       "      <td>TTATTATCTCCACTTTACAGATGAGAAAACTGTGGCTCAGATAAGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304277</th>\n",
       "      <td>chrX:99999349:G:A</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.077544</td>\n",
       "      <td>0.012644</td>\n",
       "      <td>-0.169617</td>\n",
       "      <td>0.142693</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.057932</td>\n",
       "      <td>...</td>\n",
       "      <td>737.166667</td>\n",
       "      <td>911.4</td>\n",
       "      <td>1702</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1136</td>\n",
       "      <td>508.166667</td>\n",
       "      <td>629.4</td>\n",
       "      <td>1341</td>\n",
       "      <td>AGCTAGAGTGCAGTGGCGCGATCTCGGCTCACTGAAAACTCCTCCT...</td>\n",
       "      <td>AGCTAGAGTGCAGTGGCGCGATCTCGGCTCACTGAAAACTCCTCCT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>304278 rows × 80 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0                   Variant active.any emVar.any allele1.log2FC_A549  \\\n",
       "0       chr10:100025816:G:A       True     False                 NaN   \n",
       "1       chr10:100029561:C:T       True      True                 NaN   \n",
       "2       chr10:100098716:G:T       True      True                 NaN   \n",
       "3       chr10:100109592:T:C      False     False                 NaN   \n",
       "4       chr10:100141027:T:C      False     False                 NaN   \n",
       "...                     ...        ...       ...                 ...   \n",
       "304273    chrX:99995091:A:G      False     False                 NaN   \n",
       "304274    chrX:99995933:C:G      False     False                 NaN   \n",
       "304275    chrX:99996102:A:G      False     False                 NaN   \n",
       "304276    chrX:99998829:C:T      False     False                 NaN   \n",
       "304277    chrX:99999349:G:A      False     False                 NaN   \n",
       "\n",
       "0      allele1.log2FC_HEPG2 allele1.log2FC_K562 allele1.log2FC_SKNSH  \\\n",
       "0                  1.623685            2.044789              1.22237   \n",
       "1                  3.390524            0.296395             3.402503   \n",
       "2                  1.174309            0.853279             1.283911   \n",
       "3                 -0.007378            0.301644             0.592409   \n",
       "4                 -0.238128            -0.00355            -0.288184   \n",
       "...                     ...                 ...                  ...   \n",
       "304273             0.171016             0.41844            -0.541239   \n",
       "304274             0.433557             0.23761             0.368939   \n",
       "304275              0.09405            0.310849             0.070434   \n",
       "304276            -0.496841            0.125072             0.011764   \n",
       "304277            -0.077544            0.012644            -0.169617   \n",
       "\n",
       "0      allele1.log2FC_HCT116 allele1.log2FC.SE_A549 allele1.log2FC.SE_HEPG2  \\\n",
       "0                   1.280575                    NaN                0.155521   \n",
       "1                   2.162766                    NaN                0.088808   \n",
       "2                   1.000218                    NaN                0.071422   \n",
       "3                   -0.03329                    NaN                0.139805   \n",
       "4                   0.041701                    NaN                0.076976   \n",
       "...                      ...                    ...                     ...   \n",
       "304273              0.296166                    NaN                0.136126   \n",
       "304274             -0.133435                    NaN                0.064148   \n",
       "304275              0.048445                    NaN                0.060897   \n",
       "304276             -0.355709                    NaN                0.132722   \n",
       "304277              0.142693                    NaN                0.057932   \n",
       "\n",
       "0       ... mean_RNA_ref_K562 mean_RNA_ref_SKNSH mean_RNA_ref_HCT116  \\\n",
       "0       ...        593.166667              305.4                 459   \n",
       "1       ...        665.166667               5250              3554.4   \n",
       "2       ...       1131.666667               2423                2850   \n",
       "3       ...        601.666667              643.4               718.6   \n",
       "4       ...        361.666667                417               731.2   \n",
       "...     ...               ...                ...                 ...   \n",
       "304273  ...        390.166667                281                 794   \n",
       "304274  ...        562.666667              884.2               948.6   \n",
       "304275  ...        953.333333               1182              1679.4   \n",
       "304276  ...        123.666667              207.4                 265   \n",
       "304277  ...        737.166667              911.4                1702   \n",
       "\n",
       "0      mean_RNA_alt_A549 mean_RNA_alt_HEPG2 mean_RNA_alt_K562  \\\n",
       "0                    NaN              719.6               796   \n",
       "1                    NaN             2298.8        439.333333   \n",
       "2                    NaN             1243.4        374.333333   \n",
       "3                    NaN              695.2             522.5   \n",
       "4                    NaN              627.8        374.666667   \n",
       "...                  ...                ...               ...   \n",
       "304273               NaN              211.6               226   \n",
       "304274               NaN             1853.8        657.833333   \n",
       "304275               NaN               1936        908.166667   \n",
       "304276               NaN              306.4        118.166667   \n",
       "304277               NaN               1136        508.166667   \n",
       "\n",
       "0      mean_RNA_alt_SKNSH mean_RNA_alt_HCT116  \\\n",
       "0                   599.4               656.2   \n",
       "1                  1873.8              1458.4   \n",
       "2                  1114.2              1060.4   \n",
       "3                   802.6                 754   \n",
       "4                   333.2               660.4   \n",
       "...                   ...                 ...   \n",
       "304273              140.2               320.4   \n",
       "304274             1382.6              1364.2   \n",
       "304275             1137.8              1626.6   \n",
       "304276              206.8               402.6   \n",
       "304277              629.4                1341   \n",
       "\n",
       "0                                          Allele 1 Oligo  \\\n",
       "0       TCTGACCCCTCCCACGCTTCCTTTCCAGACTGAAACACTGTCTTCT...   \n",
       "1       GCAACACAGAGGGTTCGGAAACGTGGGCTCTGCCTGCTGCCAATCA...   \n",
       "2       CCCCCAGCAAACACGGATCTGCTCTCTGTCACTATCGATTAGCTTG...   \n",
       "3       GATCTCAGGCATCCTCACTTCGTGCCGAGAGGAGGCTCTTAAACAT...   \n",
       "4       CTCCTTTACAGGGAACATTTGCCTGGAGAACAGCTGCACCCTTGTG...   \n",
       "...                                                   ...   \n",
       "304273  TAATTTATTCAGGAAAAATCTGAGGTCACTAACATTTACTAGTGCT...   \n",
       "304274  AGAGAGACTGACATCTCACTCCCAGGCTGGAGGCTTGCTGCACTAC...   \n",
       "304275  AAGCCTGGTTTAAATTGCTCATAATGGCAGAGCAAAGAAAGAGATC...   \n",
       "304276  TTATTATCTCCACTTTACAGATGAGAAAACTGTGGCTCAGATAAGT...   \n",
       "304277  AGCTAGAGTGCAGTGGCGCGATCTCGGCTCACTGAAAACTCCTCCT...   \n",
       "\n",
       "0                                          Allele 2 Oligo  \n",
       "0       TCTGACCCCTCCCACGCTTCCTTTCCAGACTGAAACACTGTCTTCT...  \n",
       "1       GCAACACAGAGGGTTCGGAAACGTGGGCTCTGCCTGCTGCCAATCA...  \n",
       "2       CCCCCAGCAAACACGGATCTGCTCTCTGTCACTATCGATTAGCTTG...  \n",
       "3       GATCTCAGGCATCCTCACTTCGTGCCGAGAGGAGGCTCTTAAACAT...  \n",
       "4       CTCCTTTACAGGGAACATTTGCCTGGAGAACAGCTGCACCCTTGTG...  \n",
       "...                                                   ...  \n",
       "304273  TAATTTATTCAGGAAAAATCTGAGGTCACTAACATTTACTAGTGCT...  \n",
       "304274  AGAGAGACTGACATCTCACTCCCAGGCTGGAGGCTTGCTGCACTAC...  \n",
       "304275  AAGCCTGGTTTAAATTGCTCATAATGGCAGAGCAAAGAAAGAGATC...  \n",
       "304276  TTATTATCTCCACTTTACAGATGAGAAAACTGTGGCTCAGATAAGT...  \n",
       "304277  AGCTAGAGTGCAGTGGCGCGATCTCGGCTCACTGAAAACTCCTCCT...  \n",
       "\n",
       "[304278 rows x 80 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df = pd.merge(df4, df3, on='Variant', how='inner')\n",
    "combined_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_df.to_csv(\"SirajMPRA_raw\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2053304/1039314259.py:1: DtypeWarning: Columns (28,43) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  combined_df = pd.read_csv('SirajMPRA_raw.csv')\n"
     ]
    }
   ],
   "source": [
    "combined_df = pd.read_csv('SirajMPRA_raw.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2053304/2622118777.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_allele_1['is_ref'] = True\n",
      "/tmp/ipykernel_2053304/2622118777.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_allele_2['is_ref'] = False\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>seq</th>\n",
       "      <th>A549</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>SKNSH</th>\n",
       "      <th>HCT116</th>\n",
       "      <th>is_ref</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100025816:G:A</td>\n",
       "      <td>TCTGACCCCTCCCACGCTTCCTTTCCAGACTGAAACACTGTCTTCT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.623685</td>\n",
       "      <td>2.044789</td>\n",
       "      <td>1.222370</td>\n",
       "      <td>1.280575</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100029561:C:T</td>\n",
       "      <td>GCAACACAGAGGGTTCGGAAACGTGGGCTCTGCCTGCTGCCAATCA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.390524</td>\n",
       "      <td>0.296395</td>\n",
       "      <td>3.402503</td>\n",
       "      <td>2.162766</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100098716:G:T</td>\n",
       "      <td>CCCCCAGCAAACACGGATCTGCTCTCTGTCACTATCGATTAGCTTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.174309</td>\n",
       "      <td>0.853279</td>\n",
       "      <td>1.283911</td>\n",
       "      <td>1.000218</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100109592:T:C</td>\n",
       "      <td>GATCTCAGGCATCCTCACTTCGTGCCGAGAGGAGGCTCTTAAACAT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.007378</td>\n",
       "      <td>0.301644</td>\n",
       "      <td>0.592409</td>\n",
       "      <td>-0.033290</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100141027:T:C</td>\n",
       "      <td>CTCCTTTACAGGGAACATTTGCCTGGAGAACAGCTGCACCCTTGTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.238128</td>\n",
       "      <td>-0.003550</td>\n",
       "      <td>-0.288184</td>\n",
       "      <td>0.041701</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608551</th>\n",
       "      <td>chrX:99995091:A:G</td>\n",
       "      <td>TAATTTATTCAGGAAAAATCTGAGGTCACTAACATTTACTAGTGCT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.188583</td>\n",
       "      <td>0.693828</td>\n",
       "      <td>-0.373362</td>\n",
       "      <td>0.203649</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608552</th>\n",
       "      <td>chrX:99995933:C:G</td>\n",
       "      <td>AGAGAGACTGACATCTCACTCCCAGGCTGGAGGCTTGCTGCACTAC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.402471</td>\n",
       "      <td>-0.325169</td>\n",
       "      <td>0.395378</td>\n",
       "      <td>0.076269</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608553</th>\n",
       "      <td>chrX:99996102:A:G</td>\n",
       "      <td>AAGCCTGGTTTAAATTGCTCATAATGGCAGAGCAAAGAAAGAGATC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.174034</td>\n",
       "      <td>0.108281</td>\n",
       "      <td>-0.039556</td>\n",
       "      <td>0.019372</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608554</th>\n",
       "      <td>chrX:99998829:C:T</td>\n",
       "      <td>TTATTATCTCCACTTTACAGATGAGAAAACTGTGGCTCAGATAAGT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.199757</td>\n",
       "      <td>0.231536</td>\n",
       "      <td>-0.198834</td>\n",
       "      <td>0.298371</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608555</th>\n",
       "      <td>chrX:99999349:G:A</td>\n",
       "      <td>AGCTAGAGTGCAGTGGCGCGATCTCGGCTCACTGAAAACTCCTCCT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.142980</td>\n",
       "      <td>-0.222729</td>\n",
       "      <td>-0.242416</td>\n",
       "      <td>0.482251</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>608556 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                description  \\\n",
       "0       chr10:100025816:G:A   \n",
       "1       chr10:100029561:C:T   \n",
       "2       chr10:100098716:G:T   \n",
       "3       chr10:100109592:T:C   \n",
       "4       chr10:100141027:T:C   \n",
       "...                     ...   \n",
       "608551    chrX:99995091:A:G   \n",
       "608552    chrX:99995933:C:G   \n",
       "608553    chrX:99996102:A:G   \n",
       "608554    chrX:99998829:C:T   \n",
       "608555    chrX:99999349:G:A   \n",
       "\n",
       "                                                      seq  A549     HepG2  \\\n",
       "0       TCTGACCCCTCCCACGCTTCCTTTCCAGACTGAAACACTGTCTTCT...   NaN  1.623685   \n",
       "1       GCAACACAGAGGGTTCGGAAACGTGGGCTCTGCCTGCTGCCAATCA...   NaN  3.390524   \n",
       "2       CCCCCAGCAAACACGGATCTGCTCTCTGTCACTATCGATTAGCTTG...   NaN  1.174309   \n",
       "3       GATCTCAGGCATCCTCACTTCGTGCCGAGAGGAGGCTCTTAAACAT...   NaN -0.007378   \n",
       "4       CTCCTTTACAGGGAACATTTGCCTGGAGAACAGCTGCACCCTTGTG...   NaN -0.238128   \n",
       "...                                                   ...   ...       ...   \n",
       "608551  TAATTTATTCAGGAAAAATCTGAGGTCACTAACATTTACTAGTGCT...   NaN -0.188583   \n",
       "608552  AGAGAGACTGACATCTCACTCCCAGGCTGGAGGCTTGCTGCACTAC...   NaN  0.402471   \n",
       "608553  AAGCCTGGTTTAAATTGCTCATAATGGCAGAGCAAAGAAAGAGATC...   NaN  0.174034   \n",
       "608554  TTATTATCTCCACTTTACAGATGAGAAAACTGTGGCTCAGATAAGT...   NaN -0.199757   \n",
       "608555  AGCTAGAGTGCAGTGGCGCGATCTCGGCTCACTGAAAACTCCTCCT...   NaN  0.142980   \n",
       "\n",
       "            K562     SKNSH    HCT116  is_ref  \n",
       "0       2.044789  1.222370  1.280575    True  \n",
       "1       0.296395  3.402503  2.162766    True  \n",
       "2       0.853279  1.283911  1.000218    True  \n",
       "3       0.301644  0.592409 -0.033290    True  \n",
       "4      -0.003550 -0.288184  0.041701    True  \n",
       "...          ...       ...       ...     ...  \n",
       "608551  0.693828 -0.373362  0.203649   False  \n",
       "608552 -0.325169  0.395378  0.076269   False  \n",
       "608553  0.108281 -0.039556  0.019372   False  \n",
       "608554  0.231536 -0.198834  0.298371   False  \n",
       "608555 -0.222729 -0.242416  0.482251   False  \n",
       "\n",
       "[608556 rows x 8 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_allele_1 = combined_df[['Variant', 'Allele 1 Oligo', 'allele1.log2FC_A549', 'allele1.log2FC_HEPG2', 'allele1.log2FC_K562', 'allele1.log2FC_SKNSH', 'allele1.log2FC_HCT116']]\n",
    "df_allele_2 = combined_df[['Variant', 'Allele 2 Oligo', 'allele2.log2FC_A549', 'allele2.log2FC_HEPG2', 'allele2.log2FC_K562', 'allele2.log2FC_SKNSH', 'allele2.log2FC_HCT116']]\n",
    "df_allele_1['is_ref'] = True\n",
    "df_allele_2['is_ref'] = False\n",
    "\n",
    "df_allele_1 = df_allele_1.rename(columns={'Variant': 'description', 'Allele 1 Oligo': 'seq', 'allele1.log2FC_A549': 'A549', 'allele1.log2FC_HEPG2': 'HepG2', 'allele1.log2FC_K562': 'K562', 'allele1.log2FC_SKNSH': 'SKNSH', 'allele1.log2FC_HCT116': 'HCT116'})\n",
    "df_allele_2 = df_allele_2.rename(columns={'Variant': 'description', 'Allele 2 Oligo': 'seq', 'allele2.log2FC_A549': 'A549', 'allele2.log2FC_HEPG2': 'HepG2', 'allele2.log2FC_K562': 'K562', 'allele2.log2FC_SKNSH': 'SKNSH', 'allele2.log2FC_HCT116': 'HCT116'})\n",
    "\n",
    "df = pd.concat([df_allele_1, df_allele_2], axis=0)\n",
    "df = df.reset_index(drop=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>seq</th>\n",
       "      <th>A549</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>SKNSH</th>\n",
       "      <th>HCT116</th>\n",
       "      <th>is_ref</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.667516</td>\n",
       "      <td>-0.704147</td>\n",
       "      <td>0.538641</td>\n",
       "      <td>0.159069</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.415984</td>\n",
       "      <td>0.554099</td>\n",
       "      <td>0.441904</td>\n",
       "      <td>-0.288019</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:63697:T:C</td>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.269937</td>\n",
       "      <td>-0.391222</td>\n",
       "      <td>0.752052</td>\n",
       "      <td>1.100437</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:63697:T:C</td>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.151995</td>\n",
       "      <td>-0.659023</td>\n",
       "      <td>0.204721</td>\n",
       "      <td>1.050619</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:64764:C:T</td>\n",
       "      <td>AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.211076</td>\n",
       "      <td>-0.007174</td>\n",
       "      <td>0.499376</td>\n",
       "      <td>0.257572</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>64764</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608551</th>\n",
       "      <td>chrX:155233098:T:C</td>\n",
       "      <td>ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.615976</td>\n",
       "      <td>1.927090</td>\n",
       "      <td>0.798263</td>\n",
       "      <td>1.035440</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233098</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608552</th>\n",
       "      <td>chrX:155233794:A:G</td>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.409758</td>\n",
       "      <td>0.556338</td>\n",
       "      <td>2.843295</td>\n",
       "      <td>1.495515</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608553</th>\n",
       "      <td>chrX:155233794:A:G</td>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.894634</td>\n",
       "      <td>-0.072647</td>\n",
       "      <td>1.973661</td>\n",
       "      <td>0.794848</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608554</th>\n",
       "      <td>chrX:155234730:T:C</td>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312172</td>\n",
       "      <td>0.165634</td>\n",
       "      <td>-0.041383</td>\n",
       "      <td>0.444472</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608555</th>\n",
       "      <td>chrX:155234730:T:C</td>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.370818</td>\n",
       "      <td>0.009694</td>\n",
       "      <td>-0.035569</td>\n",
       "      <td>0.407723</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>608556 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               description                                                seq  \\\n",
       "0           chr1:14677:G:A  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "1           chr1:14677:G:A  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "2           chr1:63697:T:C  CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   \n",
       "3           chr1:63697:T:C  CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   \n",
       "4           chr1:64764:C:T  AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...   \n",
       "...                    ...                                                ...   \n",
       "608551  chrX:155233098:T:C  ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...   \n",
       "608552  chrX:155233794:A:G  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   \n",
       "608553  chrX:155233794:A:G  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   \n",
       "608554  chrX:155234730:T:C  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   \n",
       "608555  chrX:155234730:T:C  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   \n",
       "\n",
       "        A549     HepG2      K562     SKNSH    HCT116  is_ref   chr        pos  \\\n",
       "0        NaN  0.667516 -0.704147  0.538641  0.159069    True  chr1      14677   \n",
       "1        NaN  0.415984  0.554099  0.441904 -0.288019   False  chr1      14677   \n",
       "2        NaN  1.269937 -0.391222  0.752052  1.100437    True  chr1      63697   \n",
       "3        NaN  1.151995 -0.659023  0.204721  1.050619   False  chr1      63697   \n",
       "4        NaN -0.211076 -0.007174  0.499376  0.257572    True  chr1      64764   \n",
       "...      ...       ...       ...       ...       ...     ...   ...        ...   \n",
       "608551   NaN  0.615976  1.927090  0.798263  1.035440   False  chrX  155233098   \n",
       "608552   NaN  1.409758  0.556338  2.843295  1.495515    True  chrX  155233794   \n",
       "608553   NaN  0.894634 -0.072647  1.973661  0.794848   False  chrX  155233794   \n",
       "608554   NaN  0.312172  0.165634 -0.041383  0.444472    True  chrX  155234730   \n",
       "608555   NaN  0.370818  0.009694 -0.035569  0.407723   False  chrX  155234730   \n",
       "\n",
       "       ref alt  \n",
       "0        G   A  \n",
       "1        G   A  \n",
       "2        T   C  \n",
       "3        T   C  \n",
       "4        C   T  \n",
       "...     ..  ..  \n",
       "608551   T   C  \n",
       "608552   A   G  \n",
       "608553   A   G  \n",
       "608554   T   C  \n",
       "608555   T   C  \n",
       "\n",
       "[608556 rows x 12 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['chr', 'pos', 'ref', 'alt']] = df['description'].str.split(':', expand=True)\n",
    "df['pos'] = df['pos'].astype(int)\n",
    "df = df.sort_values(by=['chr', 'pos']).reset_index(drop=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"SirajMPRA_609k.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>seq</th>\n",
       "      <th>A549</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>SKNSH</th>\n",
       "      <th>HCT116</th>\n",
       "      <th>is_ref</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.667516</td>\n",
       "      <td>-0.704147</td>\n",
       "      <td>0.538641</td>\n",
       "      <td>0.159069</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.415984</td>\n",
       "      <td>0.554099</td>\n",
       "      <td>0.441904</td>\n",
       "      <td>-0.288019</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:63697:T:C</td>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.269937</td>\n",
       "      <td>-0.391222</td>\n",
       "      <td>0.752052</td>\n",
       "      <td>1.100437</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:63697:T:C</td>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.151995</td>\n",
       "      <td>-0.659023</td>\n",
       "      <td>0.204721</td>\n",
       "      <td>1.050619</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:64764:C:T</td>\n",
       "      <td>AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.211076</td>\n",
       "      <td>-0.007174</td>\n",
       "      <td>0.499376</td>\n",
       "      <td>0.257572</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>64764</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608551</th>\n",
       "      <td>chrX:155233098:T:C</td>\n",
       "      <td>ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.615976</td>\n",
       "      <td>1.927090</td>\n",
       "      <td>0.798263</td>\n",
       "      <td>1.035440</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233098</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608552</th>\n",
       "      <td>chrX:155233794:A:G</td>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.409758</td>\n",
       "      <td>0.556338</td>\n",
       "      <td>2.843295</td>\n",
       "      <td>1.495515</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608553</th>\n",
       "      <td>chrX:155233794:A:G</td>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.894634</td>\n",
       "      <td>-0.072647</td>\n",
       "      <td>1.973661</td>\n",
       "      <td>0.794848</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608554</th>\n",
       "      <td>chrX:155234730:T:C</td>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312172</td>\n",
       "      <td>0.165634</td>\n",
       "      <td>-0.041383</td>\n",
       "      <td>0.444472</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608555</th>\n",
       "      <td>chrX:155234730:T:C</td>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.370818</td>\n",
       "      <td>0.009694</td>\n",
       "      <td>-0.035569</td>\n",
       "      <td>0.407723</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>608556 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               description                                                seq  \\\n",
       "0           chr1:14677:G:A  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "1           chr1:14677:G:A  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "2           chr1:63697:T:C  CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   \n",
       "3           chr1:63697:T:C  CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   \n",
       "4           chr1:64764:C:T  AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...   \n",
       "...                    ...                                                ...   \n",
       "608551  chrX:155233098:T:C  ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...   \n",
       "608552  chrX:155233794:A:G  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   \n",
       "608553  chrX:155233794:A:G  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   \n",
       "608554  chrX:155234730:T:C  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   \n",
       "608555  chrX:155234730:T:C  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   \n",
       "\n",
       "        A549     HepG2      K562     SKNSH    HCT116  is_ref   chr        pos  \\\n",
       "0        NaN  0.667516 -0.704147  0.538641  0.159069    True  chr1      14677   \n",
       "1        NaN  0.415984  0.554099  0.441904 -0.288019   False  chr1      14677   \n",
       "2        NaN  1.269937 -0.391222  0.752052  1.100437    True  chr1      63697   \n",
       "3        NaN  1.151995 -0.659023  0.204721  1.050619   False  chr1      63697   \n",
       "4        NaN -0.211076 -0.007174  0.499376  0.257572    True  chr1      64764   \n",
       "...      ...       ...       ...       ...       ...     ...   ...        ...   \n",
       "608551   NaN  0.615976  1.927090  0.798263  1.035440   False  chrX  155233098   \n",
       "608552   NaN  1.409758  0.556338  2.843295  1.495515    True  chrX  155233794   \n",
       "608553   NaN  0.894634 -0.072647  1.973661  0.794848   False  chrX  155233794   \n",
       "608554   NaN  0.312172  0.165634 -0.041383  0.444472    True  chrX  155234730   \n",
       "608555   NaN  0.370818  0.009694 -0.035569  0.407723   False  chrX  155234730   \n",
       "\n",
       "       ref alt  \n",
       "0        G   A  \n",
       "1        G   A  \n",
       "2        T   C  \n",
       "3        T   C  \n",
       "4        C   T  \n",
       "...     ..  ..  \n",
       "608551   T   C  \n",
       "608552   A   G  \n",
       "608553   A   G  \n",
       "608554   T   C  \n",
       "608555   T   C  \n",
       "\n",
       "[608556 rows x 12 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(\"SirajMPRA_609k.csv\")\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "description         0\n",
      "seq                 0\n",
      "A549           356964\n",
      "HepG2            2026\n",
      "K562             2704\n",
      "SKNSH            2888\n",
      "HCT116         238340\n",
      "is_ref              0\n",
      "chr                 0\n",
      "pos                 0\n",
      "ref                 0\n",
      "alt                 0\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(pd.isna(df).sum(0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>seq</th>\n",
       "      <th>A549</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>SKNSH</th>\n",
       "      <th>HCT116</th>\n",
       "      <th>is_ref</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.667516</td>\n",
       "      <td>-0.704147</td>\n",
       "      <td>0.538641</td>\n",
       "      <td>0.159069</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.415984</td>\n",
       "      <td>0.554099</td>\n",
       "      <td>0.441904</td>\n",
       "      <td>-0.288019</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:63697:T:C</td>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.269937</td>\n",
       "      <td>-0.391222</td>\n",
       "      <td>0.752052</td>\n",
       "      <td>1.100437</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:63697:T:C</td>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.151995</td>\n",
       "      <td>-0.659023</td>\n",
       "      <td>0.204721</td>\n",
       "      <td>1.050619</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:64764:C:T</td>\n",
       "      <td>AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.211076</td>\n",
       "      <td>-0.007174</td>\n",
       "      <td>0.499376</td>\n",
       "      <td>0.257572</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>64764</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585600</th>\n",
       "      <td>chrX:155233098:T:C</td>\n",
       "      <td>ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.615976</td>\n",
       "      <td>1.927090</td>\n",
       "      <td>0.798263</td>\n",
       "      <td>1.035440</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233098</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585601</th>\n",
       "      <td>chrX:155233794:A:G</td>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.409758</td>\n",
       "      <td>0.556338</td>\n",
       "      <td>2.843295</td>\n",
       "      <td>1.495515</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585602</th>\n",
       "      <td>chrX:155233794:A:G</td>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.894634</td>\n",
       "      <td>-0.072647</td>\n",
       "      <td>1.973661</td>\n",
       "      <td>0.794848</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585603</th>\n",
       "      <td>chrX:155234730:T:C</td>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312172</td>\n",
       "      <td>0.165634</td>\n",
       "      <td>-0.041383</td>\n",
       "      <td>0.444472</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585604</th>\n",
       "      <td>chrX:155234730:T:C</td>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.370818</td>\n",
       "      <td>0.009694</td>\n",
       "      <td>-0.035569</td>\n",
       "      <td>0.407723</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>585605 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               description                                                seq  \\\n",
       "0           chr1:14677:G:A  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "1           chr1:14677:G:A  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "2           chr1:63697:T:C  CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   \n",
       "3           chr1:63697:T:C  CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   \n",
       "4           chr1:64764:C:T  AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...   \n",
       "...                    ...                                                ...   \n",
       "585600  chrX:155233098:T:C  ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...   \n",
       "585601  chrX:155233794:A:G  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   \n",
       "585602  chrX:155233794:A:G  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   \n",
       "585603  chrX:155234730:T:C  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   \n",
       "585604  chrX:155234730:T:C  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   \n",
       "\n",
       "        A549     HepG2      K562     SKNSH    HCT116  is_ref   chr        pos  \\\n",
       "0        NaN  0.667516 -0.704147  0.538641  0.159069    True  chr1      14677   \n",
       "1        NaN  0.415984  0.554099  0.441904 -0.288019   False  chr1      14677   \n",
       "2        NaN  1.269937 -0.391222  0.752052  1.100437    True  chr1      63697   \n",
       "3        NaN  1.151995 -0.659023  0.204721  1.050619   False  chr1      63697   \n",
       "4        NaN -0.211076 -0.007174  0.499376  0.257572    True  chr1      64764   \n",
       "...      ...       ...       ...       ...       ...     ...   ...        ...   \n",
       "585600   NaN  0.615976  1.927090  0.798263  1.035440   False  chrX  155233098   \n",
       "585601   NaN  1.409758  0.556338  2.843295  1.495515    True  chrX  155233794   \n",
       "585602   NaN  0.894634 -0.072647  1.973661  0.794848   False  chrX  155233794   \n",
       "585603   NaN  0.312172  0.165634 -0.041383  0.444472    True  chrX  155234730   \n",
       "585604   NaN  0.370818  0.009694 -0.035569  0.407723   False  chrX  155234730   \n",
       "\n",
       "       ref alt  \n",
       "0        G   A  \n",
       "1        G   A  \n",
       "2        T   C  \n",
       "3        T   C  \n",
       "4        C   T  \n",
       "...     ..  ..  \n",
       "585600   T   C  \n",
       "585601   A   G  \n",
       "585602   A   G  \n",
       "585603   T   C  \n",
       "585604   T   C  \n",
       "\n",
       "[585605 rows x 12 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df[df['seq'].str.len() == 200]\n",
    "# df = df[~df[['HepG2', 'K562', 'SKNSH']].isna().any(axis=1)]\n",
    "df = df.reset_index(drop=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('SirajMPRA_total.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1850705/854393138.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['mean_2'] = df[['HepG2', 'K562']].mean(axis=1)\n",
      "/tmp/ipykernel_1850705/854393138.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['mean_3'] = df[['HepG2', 'K562', 'SKNSH']].mean(axis=1)\n"
     ]
    }
   ],
   "source": [
    "# df['mean_2'] = df[['HepG2', 'K562']].mean(axis=1)\n",
    "# df['mean_3'] = df[['HepG2', 'K562', 'SKNSH']].mean(axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  5%|▍         | 15103/304278 [00:00<00:07, 38331.86it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 8151\n",
      "Cannot convert line 8198\n",
      "Cannot convert line 8212\n",
      "Cannot convert line 8359\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 13%|█▎        | 40137/304278 [00:01<00:06, 40865.94it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 32334\n",
      "Cannot convert line 32345\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 20%|██        | 61393/304278 [00:01<00:06, 39565.94it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 55226\n",
      "Cannot convert line 55249\n",
      "Cannot convert line 55291\n",
      "Cannot convert line 55299\n",
      "Cannot convert line 55312\n",
      "Cannot convert line 55840\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 33%|███▎      | 99617/304278 [00:02<00:05, 39598.40it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 90946\n",
      "Cannot convert line 91033\n",
      "Cannot convert line 92230\n",
      "Cannot convert line 96730\n",
      "Cannot convert line 96732\n",
      "Cannot convert line 97867\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 35%|███▌      | 107882/304278 [00:02<00:04, 40431.90it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 100779\n",
      "Cannot convert line 100780\n",
      "Cannot convert line 100781\n",
      "Cannot convert line 100787\n",
      "Cannot convert line 107324\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 40%|████      | 122981/304278 [00:03<00:06, 28472.98it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 116388\n",
      "Cannot convert line 116414\n",
      "Cannot convert line 116481\n",
      "Cannot convert line 120369\n",
      "Cannot convert line 120459\n",
      "Cannot convert line 120547\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 43%|████▎     | 130897/304278 [00:03<00:05, 32917.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 126416\n",
      "Cannot convert line 126765\n",
      "Cannot convert line 126834\n",
      "Cannot convert line 126889\n",
      "Cannot convert line 126893\n",
      "Cannot convert line 127164\n",
      "Cannot convert line 127254\n",
      "Cannot convert line 127259\n",
      "Cannot convert line 127263\n",
      "Cannot convert line 127361\n",
      "Cannot convert line 127395\n",
      "Cannot convert line 127531\n",
      "Cannot convert line 127535\n",
      "Cannot convert line 127569\n",
      "Cannot convert line 127684\n",
      "Cannot convert line 127710\n",
      "Cannot convert line 129582\n",
      "Cannot convert line 129583\n",
      "Cannot convert line 129585\n",
      "Cannot convert line 129594\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 47%|████▋     | 142746/304278 [00:03<00:04, 36444.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 136145\n",
      "Cannot convert line 136250\n",
      "Cannot convert line 136251\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 53%|█████▎    | 159771/304278 [00:04<00:03, 40032.82it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 154910\n",
      "Cannot convert line 155151\n",
      "Cannot convert line 155154\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 57%|█████▋    | 172155/304278 [00:04<00:03, 39723.96it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 165160\n",
      "Cannot convert line 165261\n",
      "Cannot convert line 165325\n",
      "Cannot convert line 166012\n",
      "Cannot convert line 169496\n",
      "Cannot convert line 169520\n",
      "Cannot convert line 170549\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 66%|██████▋   | 201691/304278 [00:05<00:02, 38636.88it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 194180\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 72%|███████▏  | 217942/304278 [00:05<00:02, 38652.47it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 211182\n",
      "Cannot convert line 215657\n",
      "Cannot convert line 216604\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 88%|████████▊ | 267204/304278 [00:07<00:00, 39549.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 260152\n",
      "Cannot convert line 260199\n",
      "Cannot convert line 261403\n",
      "Cannot convert line 261441\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 91%|█████████ | 275570/304278 [00:07<00:00, 40304.54it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 269859\n",
      "Cannot convert line 270544\n",
      "Cannot convert line 270549\n",
      "Cannot convert line 270555\n",
      "Cannot convert line 271046\n",
      "Cannot convert line 271078\n",
      "Cannot convert line 271109\n",
      "Cannot convert line 271118\n",
      "Cannot convert line 271167\n",
      "Cannot convert line 271227\n",
      "Cannot convert line 271254\n",
      "Cannot convert line 271278\n",
      "Cannot convert line 271323\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 95%|█████████▍| 287878/304278 [00:07<00:00, 39963.42it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 279685\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 96%|█████████▌| 291992/304278 [00:07<00:00, 40307.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 291967\n",
      "Cannot convert line 291968\n",
      "Cannot convert line 291969\n",
      "Cannot convert line 291986\n",
      "Cannot convert line 291988\n",
      "Cannot convert line 291996\n",
      "Cannot convert line 292016\n",
      "Cannot convert line 292026\n",
      "Cannot convert line 292035\n",
      "Cannot convert line 292046\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 99%|█████████▊| 299773/304278 [00:08<00:00, 26101.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 292687\n",
      "Cannot convert line 300822\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 304278/304278 [00:08<00:00, 37123.22it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 301692\n",
      "Cannot convert line 301757\n",
      "Cannot convert line 301830\n",
      "Cannot convert line 302027\n",
      "Cannot convert line 302360\n",
      "Cannot convert line 302462\n",
      "Cannot convert line 303616\n"
     ]
    }
   ],
   "source": [
    "# from pyliftover import LiftOver\n",
    "# lo = LiftOver('hg19', 'hg38')\n",
    "# # lo.convert_coordinate('chr1', 1000000)\n",
    "\n",
    "# new_df = []\n",
    "# for i in tqdm(range(len(df))):\n",
    "#     info = df.iloc[i, 0]\n",
    "#     chr, pos, base1, base2 = info.split(':')\n",
    "#     pos = int(pos)\n",
    "\n",
    "#     converted = lo.convert_coordinate(chr, pos)\n",
    "#     if converted and len(converted) > 0:\n",
    "#         hg38_chr, hg38_pos = converted[0][0], converted[0][1]\n",
    "#     else:\n",
    "#         hg38_chr, hg38_pos = 'NaN', 0\n",
    "#         print(f'Cannot convert line {i}')\n",
    "    \n",
    "#     new_df.append([chr, pos, hg38_chr, hg38_pos, base1, base2])\n",
    "\n",
    "# new_df = pd.DataFrame(new_df, columns=['hg19_chr','hg19_pos','hg38_chr','hg38_pos','base1','base2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = pd.read_csv('SirajMPRA_data_concat.csv')\n",
    "d = d[d['seq'].str.len() == 200]\n",
    "d.to_csv('SirajMPRA_len200.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = pd.read_csv('SirajMPRA_len200.csv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq</th>\n",
       "      <th>A549</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>SKNSH</th>\n",
       "      <th>HCT116</th>\n",
       "      <th>is_ref</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.667516</td>\n",
       "      <td>-0.704147</td>\n",
       "      <td>0.538641</td>\n",
       "      <td>0.159069</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.415984</td>\n",
       "      <td>0.554099</td>\n",
       "      <td>0.441904</td>\n",
       "      <td>-0.288019</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.269937</td>\n",
       "      <td>-0.391222</td>\n",
       "      <td>0.752052</td>\n",
       "      <td>1.100437</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.151995</td>\n",
       "      <td>-0.659023</td>\n",
       "      <td>0.204721</td>\n",
       "      <td>1.050619</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.211076</td>\n",
       "      <td>-0.007174</td>\n",
       "      <td>0.499376</td>\n",
       "      <td>0.257572</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>64764</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585600</th>\n",
       "      <td>ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.615976</td>\n",
       "      <td>1.927090</td>\n",
       "      <td>0.798263</td>\n",
       "      <td>1.035440</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233098</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585601</th>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.409758</td>\n",
       "      <td>0.556338</td>\n",
       "      <td>2.843295</td>\n",
       "      <td>1.495515</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585602</th>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.894634</td>\n",
       "      <td>-0.072647</td>\n",
       "      <td>1.973661</td>\n",
       "      <td>0.794848</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585603</th>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312172</td>\n",
       "      <td>0.165634</td>\n",
       "      <td>-0.041383</td>\n",
       "      <td>0.444472</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585604</th>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.370818</td>\n",
       "      <td>0.009694</td>\n",
       "      <td>-0.035569</td>\n",
       "      <td>0.407723</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>585605 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      seq  A549     HepG2  \\\n",
       "0       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   NaN  0.667516   \n",
       "1       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   NaN  0.415984   \n",
       "2       CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   NaN  1.269937   \n",
       "3       CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   NaN  1.151995   \n",
       "4       AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...   NaN -0.211076   \n",
       "...                                                   ...   ...       ...   \n",
       "585600  ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...   NaN  0.615976   \n",
       "585601  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   NaN  1.409758   \n",
       "585602  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   NaN  0.894634   \n",
       "585603  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   NaN  0.312172   \n",
       "585604  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   NaN  0.370818   \n",
       "\n",
       "            K562     SKNSH    HCT116  is_ref   chr        pos ref alt  \n",
       "0      -0.704147  0.538641  0.159069    True  chr1      14677   G   A  \n",
       "1       0.554099  0.441904 -0.288019   False  chr1      14677   G   A  \n",
       "2      -0.391222  0.752052  1.100437    True  chr1      63697   T   C  \n",
       "3      -0.659023  0.204721  1.050619   False  chr1      63697   T   C  \n",
       "4      -0.007174  0.499376  0.257572    True  chr1      64764   C   T  \n",
       "...          ...       ...       ...     ...   ...        ...  ..  ..  \n",
       "585600  1.927090  0.798263  1.035440   False  chrX  155233098   T   C  \n",
       "585601  0.556338  2.843295  1.495515    True  chrX  155233794   A   G  \n",
       "585602 -0.072647  1.973661  0.794848   False  chrX  155233794   A   G  \n",
       "585603  0.165634 -0.041383  0.444472    True  chrX  155234730   T   C  \n",
       "585604  0.009694 -0.035569  0.407723   False  chrX  155234730   T   C  \n",
       "\n",
       "[585605 rows x 11 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "# from pyfaidx import Fasta\n",
    "# genome = Fasta('/home/hxcai/genome/hg19.fa')\n",
    "\n",
    "# seqs = []\n",
    "# for i in tqdm(range(len(d))):\n",
    "#     chr, start, end = d.iloc[i][['chr','start', 'end']]\n",
    "#     seq = genome[chr][start: end].seq.upper()\n",
    "#     seqs.append(seq)\n",
    "\n",
    "# d['seq'] = seqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1:761957:A:AT\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTCATAAGCTGAACCTGACACAATAAATTTAAAAGGGAAACTAATTTGGAAATCAGAAAACCACTAAGGAATTTGGGAATTAGGCTTCTGCTGCCCTCTCTGCTACTGACGGTCAAGGCCTCCTCATTGTATTCTGTCCTCCATATCTCTGCTGATTCCCATT\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTCATAAGCTGAACCTGACACAATAAATTTAAAAGGGAAACTAATTTGGAAATCAGAAAACCATCTAAGGAATTTGGGAATTAGGCTTCTGCTGCCCTCTCTGCTACTGACGGTCAAGGCCTCCTCATTGTATTCTGTCCTCCATATCTCTGCTGATTCCCATT\n",
      "TTTGTCCTCAGTGGGGGCATCGGATGCACGGGCACCATCATCGTGATTGACATCCTGGTGGACGTCATCCGCAGGCAGGGCGAGCCGCTCCACTCTGGGTCCCCCCGCCCTGCCCTGCTATAGCCCCACCCCTCCGGGCGACCCCACCCCTCCGGACGACCCCACCCCTCAGGACGACCCCACCCCTCAGGACGACCCCA\n",
      "199 200 200\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTCATAAGCTGAACCTGACACAATAAATTTAAAAGGGAAACTAATTTGGAAATCAGAAAACCA-CTAAGGAATTTGGGAATTAGGCTTCTGCTGCCCTCTCTGCTACTGACGGTCAAGGCCTCCTCATTGTATTCTGTCCTCCATATCTCTGCTGATTCCCATT\n",
      "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTCATAAGCTGAACCTGACACAATAAATTTAAAAGGGAAACTAATTTGGAAATCAGAAAACCATCTAAGGAATTTGGGAATTAGGCTTCTGCTGCCCTCTCTGCTACTGACGGTCAAGGCCTCCTCATTGTATTCTGTCCTCCATATCTCTGCTGATTCCCATT\n",
      "  Score=397.5\n",
      "\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTC--ATAAGCTGAACCTGACACAATAAATT-TAAAAGGGAAACTAAT--T--TGGAAATCAGAAAACCACTAAGGAATTTGGGAATTAG--GCT----TCTG-------CTGCCCT-CTCTGCTACT-G----A------CGGTC-A----A-------GG-------C-CTCCTCATTGTATTCTGTCCTCCATATCTCTG--CTGATTCCCATT\n",
      "||    |.|.|   |.||  ||    |    | |||.||  ||  ||   | |.|.|||.|| .||. |.|..|    ||  ||  |  ||||..||    |.||.|  |||.|   |||  ..||  |||    ||||       |.||||| |.|||||| | |    |      |||.| |    |       ||       | |.|||||  |.|  | |.|| |||...|||.|  | || .||||  \n",
      "TT----TGTCC---TCAG--TG----G----G-GGCATCGGAT--GC---A-CGGGCACCAT-CATCGTGATTG----AC--ATCCTGGTGGACGTC----ATCCGC--AGGCA---GGG--CGAGCCGCTCCACTCTGGGTCCCCCCGCCCTGCCCTGCTA-TAGCCCCACCCCTCCGGGCGACCCCACCCCTCCGGACGACCCCACCCCTCA--GGA--C-GACC-CCACCCCTCAGGAC-GA-CCCCA--\n",
      "  Score=204.9\n",
      "\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTC--ATAAGCTGAACCTGACACAATAAATT-TAAAAGGGAAACTAAT--T--TGGAAATCAGAAAACCATCTAAGGAATTTGGGAATTAG--GCT----TCTG-------CTGCCCT-CTCTGCTACT-G----A------CGGTC-A----A-------GG-------C-CTCCTCATTGTATTCTGTCCTCCATATCTCTG--CTGATTCCCATT\n",
      "||    |.|.|   |.||  ||    |    | |||.||  ||  ||   | |.|.|||.|| .||. |.|..|    ||  ||  |  ||||..||    |.|| .|  |||.|   |||  ..||  |||    ||||       |.||||| |.|||||| | |    |      |||.| |    |       ||       | |.|||||  |.|  | |.|| |||...|||.|  | || .||||  \n",
      "TT----TGTCC---TCAG--TG----G----G-GGCATCGGAT--GC---A-CGGGCACCAT-CATCGTGATTG----AC--ATCCTGGTGGACGTC----ATCC-GC--AGGCA---GGG--CGAGCCGCTCCACTCTGGGTCCCCCCGCCCTGCCCTGCTA-TAGCCCCACCCCTCCGGGCGACCCCACCCCTCCGGACGACCCCACCCCTCA--GGA--C-GACC-CCACCCCTCAGGAC-GA-CCCCA--\n",
      "  Score=204.4\n",
      "\n"
     ]
    }
   ],
   "source": [
    "info, seq1, seq2 = df.iloc[15]\n",
    "\n",
    "print(info)\n",
    "print(seq1)\n",
    "print(seq2)\n",
    "\n",
    "from pyfaidx import Fasta\n",
    "genome = Fasta('/home/hxcai/genome/hg19.fa')\n",
    "seq3 = genome['chr1'][1004389-100: 1004389+100].seq.upper()\n",
    "\n",
    "print(seq3)\n",
    "print(len(seq1), len(seq2), len(seq3))\n",
    "\n",
    "alignments = align.globalms(seq1, seq2, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "\n",
    "alignments = align.globalms(seq1, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "\n",
    "alignments = align.globalms(seq2, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1:759444:ATATT:A\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "200 196 200\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||    ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAA----TATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "  Score=391.2\n",
      "\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "  Score=400\n",
      "\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAA----TATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||    ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "  Score=391.2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "info, seq1, seq2 = df.iloc[10]\n",
    "\n",
    "print(info)\n",
    "print(seq1)\n",
    "print(seq2)\n",
    "\n",
    "from pyfaidx import Fasta\n",
    "genome = Fasta('/home/hxcai/genome/hg19.fa')\n",
    "seq3 = genome['chr1'][759444-100+2: 759444+100+2].seq.upper()\n",
    "\n",
    "print(seq3)\n",
    "print(len(seq1), len(seq2), len(seq3))\n",
    "\n",
    "alignments = align.globalms(seq1, seq2, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "\n",
    "alignments = align.globalms(seq1, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "\n",
    "alignments = align.globalms(seq2, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "torch",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "vscode": {
   "interpreter": {
    "hash": "8e62da015c2b8e1460a389c60847c46fcac689812463b880d1463e5f1a16b0d6"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
