{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hxcai/anaconda3/envs/torch/lib/python3.10/site-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "from Bio import pairwise2\n",
    "from Bio.pairwise2 import format_alignment, align\n",
    "\n",
    "import sys\n",
    "sys.path.append('/home/hxcai/cell_type_specific_CRE')\n",
    "from MPRA_exp.utils import *\n",
    "os.chdir('../../data/SirajMPRA')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('../../data/SirajMPRA')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 这个的染色体坐标是 hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df = pd.read_excel('media-4.xlsx')\n",
    "\n",
    "# df.columns = df.iloc[0]\n",
    "# df = df.iloc[1:].reset_index(drop=True)\n",
    "\n",
    "# # a = np.log2((df['mean_RNA_ref_HEPG2'] / df['mean_Plasmid_ref_HEPG2']).astype(float))\n",
    "# # b = df['allele1.log2FC_HEPG2']\n",
    "# # sns.scatterplot(x=a[:10000], y=b[:10000], s = 1)\n",
    "\n",
    "# select_columns = ['allele1.log2FC_A549', 'allele1.log2FC_HEPG2', 'allele1.log2FC_K562', 'allele1.log2FC_SKNSH', 'allele1.log2FC_HCT116', \n",
    "#                   'allele2.log2FC_A549', 'allele2.log2FC_HEPG2', 'allele2.log2FC_K562', 'allele2.log2FC_SKNSH', 'allele2.log2FC_HCT116']\n",
    "# df = df[['variant'] + select_columns]\n",
    "# df.to_csv('SirajMPRA_data.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>allele1.log2FC_A549</th>\n",
       "      <th>allele1.log2FC_HEPG2</th>\n",
       "      <th>allele1.log2FC_K562</th>\n",
       "      <th>allele1.log2FC_SKNSH</th>\n",
       "      <th>allele1.log2FC_HCT116</th>\n",
       "      <th>allele2.log2FC_A549</th>\n",
       "      <th>allele2.log2FC_HEPG2</th>\n",
       "      <th>allele2.log2FC_K562</th>\n",
       "      <th>allele2.log2FC_SKNSH</th>\n",
       "      <th>allele2.log2FC_HCT116</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100025816:G:A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.623685</td>\n",
       "      <td>2.044789</td>\n",
       "      <td>1.222370</td>\n",
       "      <td>1.280575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.649226</td>\n",
       "      <td>1.950421</td>\n",
       "      <td>1.807569</td>\n",
       "      <td>1.162127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100029561:C:T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.390524</td>\n",
       "      <td>0.296395</td>\n",
       "      <td>3.402503</td>\n",
       "      <td>2.162766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.610401</td>\n",
       "      <td>0.443055</td>\n",
       "      <td>2.651665</td>\n",
       "      <td>1.619869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100098716:G:T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.174309</td>\n",
       "      <td>0.853279</td>\n",
       "      <td>1.283911</td>\n",
       "      <td>1.000218</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.063949</td>\n",
       "      <td>0.748978</td>\n",
       "      <td>1.688784</td>\n",
       "      <td>0.928368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100109592:T:C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.007378</td>\n",
       "      <td>0.301644</td>\n",
       "      <td>0.592409</td>\n",
       "      <td>-0.033290</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.265591</td>\n",
       "      <td>0.059337</td>\n",
       "      <td>0.883992</td>\n",
       "      <td>0.054046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100141027:T:C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.238128</td>\n",
       "      <td>-0.003550</td>\n",
       "      <td>-0.288184</td>\n",
       "      <td>0.041701</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.391381</td>\n",
       "      <td>-0.196380</td>\n",
       "      <td>-0.657045</td>\n",
       "      <td>-0.190311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304273</th>\n",
       "      <td>chrX:99995091:A:G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.171016</td>\n",
       "      <td>0.418440</td>\n",
       "      <td>-0.541239</td>\n",
       "      <td>0.296166</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.188583</td>\n",
       "      <td>0.693828</td>\n",
       "      <td>-0.373362</td>\n",
       "      <td>0.203649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304274</th>\n",
       "      <td>chrX:99995933:C:G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.433557</td>\n",
       "      <td>0.237610</td>\n",
       "      <td>0.368939</td>\n",
       "      <td>-0.133435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.402471</td>\n",
       "      <td>-0.325169</td>\n",
       "      <td>0.395378</td>\n",
       "      <td>0.076269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304275</th>\n",
       "      <td>chrX:99996102:A:G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.094050</td>\n",
       "      <td>0.310849</td>\n",
       "      <td>0.070434</td>\n",
       "      <td>0.048445</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.174034</td>\n",
       "      <td>0.108281</td>\n",
       "      <td>-0.039556</td>\n",
       "      <td>0.019372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304276</th>\n",
       "      <td>chrX:99998829:C:T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.496841</td>\n",
       "      <td>0.125072</td>\n",
       "      <td>0.011764</td>\n",
       "      <td>-0.355709</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.199757</td>\n",
       "      <td>0.231536</td>\n",
       "      <td>-0.198834</td>\n",
       "      <td>0.298371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304277</th>\n",
       "      <td>chrX:99999349:G:A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.077544</td>\n",
       "      <td>0.012644</td>\n",
       "      <td>-0.169617</td>\n",
       "      <td>0.142693</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.142980</td>\n",
       "      <td>-0.222729</td>\n",
       "      <td>-0.242416</td>\n",
       "      <td>0.482251</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>304278 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Variant  allele1.log2FC_A549  allele1.log2FC_HEPG2  \\\n",
       "0       chr10:100025816:G:A                  NaN              1.623685   \n",
       "1       chr10:100029561:C:T                  NaN              3.390524   \n",
       "2       chr10:100098716:G:T                  NaN              1.174309   \n",
       "3       chr10:100109592:T:C                  NaN             -0.007378   \n",
       "4       chr10:100141027:T:C                  NaN             -0.238128   \n",
       "...                     ...                  ...                   ...   \n",
       "304273    chrX:99995091:A:G                  NaN              0.171016   \n",
       "304274    chrX:99995933:C:G                  NaN              0.433557   \n",
       "304275    chrX:99996102:A:G                  NaN              0.094050   \n",
       "304276    chrX:99998829:C:T                  NaN             -0.496841   \n",
       "304277    chrX:99999349:G:A                  NaN             -0.077544   \n",
       "\n",
       "        allele1.log2FC_K562  allele1.log2FC_SKNSH  allele1.log2FC_HCT116  \\\n",
       "0                  2.044789              1.222370               1.280575   \n",
       "1                  0.296395              3.402503               2.162766   \n",
       "2                  0.853279              1.283911               1.000218   \n",
       "3                  0.301644              0.592409              -0.033290   \n",
       "4                 -0.003550             -0.288184               0.041701   \n",
       "...                     ...                   ...                    ...   \n",
       "304273             0.418440             -0.541239               0.296166   \n",
       "304274             0.237610              0.368939              -0.133435   \n",
       "304275             0.310849              0.070434               0.048445   \n",
       "304276             0.125072              0.011764              -0.355709   \n",
       "304277             0.012644             -0.169617               0.142693   \n",
       "\n",
       "        allele2.log2FC_A549  allele2.log2FC_HEPG2  allele2.log2FC_K562  \\\n",
       "0                       NaN              1.649226             1.950421   \n",
       "1                       NaN              2.610401             0.443055   \n",
       "2                       NaN              1.063949             0.748978   \n",
       "3                       NaN              0.265591             0.059337   \n",
       "4                       NaN             -0.391381            -0.196380   \n",
       "...                     ...                   ...                  ...   \n",
       "304273                  NaN             -0.188583             0.693828   \n",
       "304274                  NaN              0.402471            -0.325169   \n",
       "304275                  NaN              0.174034             0.108281   \n",
       "304276                  NaN             -0.199757             0.231536   \n",
       "304277                  NaN              0.142980            -0.222729   \n",
       "\n",
       "        allele2.log2FC_SKNSH  allele2.log2FC_HCT116  \n",
       "0                   1.807569               1.162127  \n",
       "1                   2.651665               1.619869  \n",
       "2                   1.688784               0.928368  \n",
       "3                   0.883992               0.054046  \n",
       "4                  -0.657045              -0.190311  \n",
       "...                      ...                    ...  \n",
       "304273             -0.373362               0.203649  \n",
       "304274              0.395378               0.076269  \n",
       "304275             -0.039556               0.019372  \n",
       "304276             -0.198834               0.298371  \n",
       "304277             -0.242416               0.482251  \n",
       "\n",
       "[304278 rows x 11 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "value_df = pd.read_csv('SirajMPRA_data.csv')\n",
    "value_df = value_df.rename(columns={'variant': 'Variant'})\n",
    "value_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# value_df[['chr', 'pos', 'ref', 'alt']] = value_df['Variant'].str.split(':', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>allele1.log2FC_A549</th>\n",
       "      <th>allele1.log2FC_HEPG2</th>\n",
       "      <th>allele1.log2FC_K562</th>\n",
       "      <th>allele1.log2FC_SKNSH</th>\n",
       "      <th>allele1.log2FC_HCT116</th>\n",
       "      <th>allele2.log2FC_A549</th>\n",
       "      <th>allele2.log2FC_HEPG2</th>\n",
       "      <th>allele2.log2FC_K562</th>\n",
       "      <th>allele2.log2FC_SKNSH</th>\n",
       "      <th>allele2.log2FC_HCT116</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100025816:G:A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.623685</td>\n",
       "      <td>2.044789</td>\n",
       "      <td>1.222370</td>\n",
       "      <td>1.280575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.649226</td>\n",
       "      <td>1.950421</td>\n",
       "      <td>1.807569</td>\n",
       "      <td>1.162127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100029561:C:T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.390524</td>\n",
       "      <td>0.296395</td>\n",
       "      <td>3.402503</td>\n",
       "      <td>2.162766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.610401</td>\n",
       "      <td>0.443055</td>\n",
       "      <td>2.651665</td>\n",
       "      <td>1.619869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100098716:G:T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.174309</td>\n",
       "      <td>0.853279</td>\n",
       "      <td>1.283911</td>\n",
       "      <td>1.000218</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.063949</td>\n",
       "      <td>0.748978</td>\n",
       "      <td>1.688784</td>\n",
       "      <td>0.928368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100109592:T:C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.007378</td>\n",
       "      <td>0.301644</td>\n",
       "      <td>0.592409</td>\n",
       "      <td>-0.033290</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.265591</td>\n",
       "      <td>0.059337</td>\n",
       "      <td>0.883992</td>\n",
       "      <td>0.054046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100141027:T:C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.238128</td>\n",
       "      <td>-0.003550</td>\n",
       "      <td>-0.288184</td>\n",
       "      <td>0.041701</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.391381</td>\n",
       "      <td>-0.196380</td>\n",
       "      <td>-0.657045</td>\n",
       "      <td>-0.190311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304273</th>\n",
       "      <td>chrX:99995091:A:G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.171016</td>\n",
       "      <td>0.418440</td>\n",
       "      <td>-0.541239</td>\n",
       "      <td>0.296166</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.188583</td>\n",
       "      <td>0.693828</td>\n",
       "      <td>-0.373362</td>\n",
       "      <td>0.203649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304274</th>\n",
       "      <td>chrX:99995933:C:G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.433557</td>\n",
       "      <td>0.237610</td>\n",
       "      <td>0.368939</td>\n",
       "      <td>-0.133435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.402471</td>\n",
       "      <td>-0.325169</td>\n",
       "      <td>0.395378</td>\n",
       "      <td>0.076269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304275</th>\n",
       "      <td>chrX:99996102:A:G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.094050</td>\n",
       "      <td>0.310849</td>\n",
       "      <td>0.070434</td>\n",
       "      <td>0.048445</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.174034</td>\n",
       "      <td>0.108281</td>\n",
       "      <td>-0.039556</td>\n",
       "      <td>0.019372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304276</th>\n",
       "      <td>chrX:99998829:C:T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.496841</td>\n",
       "      <td>0.125072</td>\n",
       "      <td>0.011764</td>\n",
       "      <td>-0.355709</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.199757</td>\n",
       "      <td>0.231536</td>\n",
       "      <td>-0.198834</td>\n",
       "      <td>0.298371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304277</th>\n",
       "      <td>chrX:99999349:G:A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.077544</td>\n",
       "      <td>0.012644</td>\n",
       "      <td>-0.169617</td>\n",
       "      <td>0.142693</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.142980</td>\n",
       "      <td>-0.222729</td>\n",
       "      <td>-0.242416</td>\n",
       "      <td>0.482251</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>304278 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Variant  allele1.log2FC_A549  allele1.log2FC_HEPG2  \\\n",
       "0       chr10:100025816:G:A                  NaN              1.623685   \n",
       "1       chr10:100029561:C:T                  NaN              3.390524   \n",
       "2       chr10:100098716:G:T                  NaN              1.174309   \n",
       "3       chr10:100109592:T:C                  NaN             -0.007378   \n",
       "4       chr10:100141027:T:C                  NaN             -0.238128   \n",
       "...                     ...                  ...                   ...   \n",
       "304273    chrX:99995091:A:G                  NaN              0.171016   \n",
       "304274    chrX:99995933:C:G                  NaN              0.433557   \n",
       "304275    chrX:99996102:A:G                  NaN              0.094050   \n",
       "304276    chrX:99998829:C:T                  NaN             -0.496841   \n",
       "304277    chrX:99999349:G:A                  NaN             -0.077544   \n",
       "\n",
       "        allele1.log2FC_K562  allele1.log2FC_SKNSH  allele1.log2FC_HCT116  \\\n",
       "0                  2.044789              1.222370               1.280575   \n",
       "1                  0.296395              3.402503               2.162766   \n",
       "2                  0.853279              1.283911               1.000218   \n",
       "3                  0.301644              0.592409              -0.033290   \n",
       "4                 -0.003550             -0.288184               0.041701   \n",
       "...                     ...                   ...                    ...   \n",
       "304273             0.418440             -0.541239               0.296166   \n",
       "304274             0.237610              0.368939              -0.133435   \n",
       "304275             0.310849              0.070434               0.048445   \n",
       "304276             0.125072              0.011764              -0.355709   \n",
       "304277             0.012644             -0.169617               0.142693   \n",
       "\n",
       "        allele2.log2FC_A549  allele2.log2FC_HEPG2  allele2.log2FC_K562  \\\n",
       "0                       NaN              1.649226             1.950421   \n",
       "1                       NaN              2.610401             0.443055   \n",
       "2                       NaN              1.063949             0.748978   \n",
       "3                       NaN              0.265591             0.059337   \n",
       "4                       NaN             -0.391381            -0.196380   \n",
       "...                     ...                   ...                  ...   \n",
       "304273                  NaN             -0.188583             0.693828   \n",
       "304274                  NaN              0.402471            -0.325169   \n",
       "304275                  NaN              0.174034             0.108281   \n",
       "304276                  NaN             -0.199757             0.231536   \n",
       "304277                  NaN              0.142980            -0.222729   \n",
       "\n",
       "        allele2.log2FC_SKNSH  allele2.log2FC_HCT116  \n",
       "0                   1.807569               1.162127  \n",
       "1                   2.651665               1.619869  \n",
       "2                   1.688784               0.928368  \n",
       "3                   0.883992               0.054046  \n",
       "4                  -0.657045              -0.190311  \n",
       "...                      ...                    ...  \n",
       "304273             -0.373362               0.203649  \n",
       "304274              0.395378               0.076269  \n",
       "304275             -0.039556               0.019372  \n",
       "304276             -0.198834               0.298371  \n",
       "304277             -0.242416               0.482251  \n",
       "\n",
       "[304278 rows x 11 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "value_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_excel('media-3.xlsx')\n",
    "df.columns = df.iloc[0]\n",
    "df = df.iloc[1:].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Allele 1 Oligo</th>\n",
       "      <th>Allele 2 Oligo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:522224:G:A</td>\n",
       "      <td>TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...</td>\n",
       "      <td>TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:729679:C:G</td>\n",
       "      <td>TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...</td>\n",
       "      <td>TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:741397:A:G</td>\n",
       "      <td>TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...</td>\n",
       "      <td>TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:752566:G:A</td>\n",
       "      <td>AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...</td>\n",
       "      <td>AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365757</th>\n",
       "      <td>chr4:57975328:A:G</td>\n",
       "      <td>TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...</td>\n",
       "      <td>TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365758</th>\n",
       "      <td>chr20:36661369:C:A</td>\n",
       "      <td>GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...</td>\n",
       "      <td>GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365759</th>\n",
       "      <td>chr8:60683160:C:T</td>\n",
       "      <td>GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...</td>\n",
       "      <td>GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365760</th>\n",
       "      <td>chr7:31046388:G:C</td>\n",
       "      <td>CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...</td>\n",
       "      <td>CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365761</th>\n",
       "      <td>chr10:10216373:A:G</td>\n",
       "      <td>GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...</td>\n",
       "      <td>GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>365762 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0                  Variant                                     Allele 1 Oligo  \\\n",
       "0           chr1:14677:G:A  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "1          chr1:522224:G:A  TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...   \n",
       "2          chr1:729679:C:G  TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...   \n",
       "3          chr1:741397:A:G  TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...   \n",
       "4          chr1:752566:G:A  AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...   \n",
       "...                    ...                                                ...   \n",
       "365757   chr4:57975328:A:G  TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...   \n",
       "365758  chr20:36661369:C:A  GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...   \n",
       "365759   chr8:60683160:C:T  GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...   \n",
       "365760   chr7:31046388:G:C  CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...   \n",
       "365761  chr10:10216373:A:G  GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...   \n",
       "\n",
       "0                                          Allele 2 Oligo  \n",
       "0       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...  \n",
       "1       TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...  \n",
       "2       TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...  \n",
       "3       TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...  \n",
       "4       AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...  \n",
       "...                                                   ...  \n",
       "365757  TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...  \n",
       "365758  GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...  \n",
       "365759  GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...  \n",
       "365760  CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...  \n",
       "365761  GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...  \n",
       "\n",
       "[365762 rows x 3 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_df = pd.merge(df, value_df, on='Variant', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Allele 1 Oligo</th>\n",
       "      <th>Allele 2 Oligo</th>\n",
       "      <th>allele1.log2FC_A549</th>\n",
       "      <th>allele1.log2FC_HEPG2</th>\n",
       "      <th>allele1.log2FC_K562</th>\n",
       "      <th>allele1.log2FC_SKNSH</th>\n",
       "      <th>allele1.log2FC_HCT116</th>\n",
       "      <th>allele2.log2FC_A549</th>\n",
       "      <th>allele2.log2FC_HEPG2</th>\n",
       "      <th>allele2.log2FC_K562</th>\n",
       "      <th>allele2.log2FC_SKNSH</th>\n",
       "      <th>allele2.log2FC_HCT116</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.667516</td>\n",
       "      <td>-0.704147</td>\n",
       "      <td>0.538641</td>\n",
       "      <td>0.159069</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.415984</td>\n",
       "      <td>0.554099</td>\n",
       "      <td>0.441904</td>\n",
       "      <td>-0.288019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:522224:G:A</td>\n",
       "      <td>TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...</td>\n",
       "      <td>TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.627627</td>\n",
       "      <td>0.200635</td>\n",
       "      <td>-0.088036</td>\n",
       "      <td>0.771700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.316303</td>\n",
       "      <td>-0.015048</td>\n",
       "      <td>-0.231999</td>\n",
       "      <td>0.587129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:729679:C:G</td>\n",
       "      <td>TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...</td>\n",
       "      <td>TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.273272</td>\n",
       "      <td>-0.051526</td>\n",
       "      <td>-0.331275</td>\n",
       "      <td>-0.340771</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.062887</td>\n",
       "      <td>-0.699992</td>\n",
       "      <td>-0.469590</td>\n",
       "      <td>-0.604452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:741397:A:G</td>\n",
       "      <td>TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...</td>\n",
       "      <td>TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.114727</td>\n",
       "      <td>0.057421</td>\n",
       "      <td>-0.474388</td>\n",
       "      <td>0.402556</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.040862</td>\n",
       "      <td>-0.073117</td>\n",
       "      <td>0.072672</td>\n",
       "      <td>0.264375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:752566:G:A</td>\n",
       "      <td>AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...</td>\n",
       "      <td>AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.067281</td>\n",
       "      <td>-0.140224</td>\n",
       "      <td>0.091286</td>\n",
       "      <td>-0.329839</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.149188</td>\n",
       "      <td>-0.090697</td>\n",
       "      <td>0.094279</td>\n",
       "      <td>-0.148007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304273</th>\n",
       "      <td>chr4:57975328:A:G</td>\n",
       "      <td>TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...</td>\n",
       "      <td>TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...</td>\n",
       "      <td>0.764494</td>\n",
       "      <td>0.243506</td>\n",
       "      <td>0.507141</td>\n",
       "      <td>1.145670</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.539478</td>\n",
       "      <td>0.220710</td>\n",
       "      <td>0.458261</td>\n",
       "      <td>0.885200</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304274</th>\n",
       "      <td>chr20:36661369:C:A</td>\n",
       "      <td>GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...</td>\n",
       "      <td>GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...</td>\n",
       "      <td>-0.475731</td>\n",
       "      <td>-0.325048</td>\n",
       "      <td>-0.227184</td>\n",
       "      <td>-0.543515</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.350664</td>\n",
       "      <td>0.008168</td>\n",
       "      <td>-0.528934</td>\n",
       "      <td>-0.164957</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304275</th>\n",
       "      <td>chr8:60683160:C:T</td>\n",
       "      <td>GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...</td>\n",
       "      <td>GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...</td>\n",
       "      <td>-0.080497</td>\n",
       "      <td>-0.020381</td>\n",
       "      <td>-0.070786</td>\n",
       "      <td>0.394575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.151279</td>\n",
       "      <td>0.258262</td>\n",
       "      <td>-0.061669</td>\n",
       "      <td>0.296101</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304276</th>\n",
       "      <td>chr7:31046388:G:C</td>\n",
       "      <td>CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...</td>\n",
       "      <td>CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...</td>\n",
       "      <td>0.359767</td>\n",
       "      <td>1.417800</td>\n",
       "      <td>2.603072</td>\n",
       "      <td>1.690492</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.913629</td>\n",
       "      <td>2.021390</td>\n",
       "      <td>3.804445</td>\n",
       "      <td>2.135197</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304277</th>\n",
       "      <td>chr10:10216373:A:G</td>\n",
       "      <td>GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...</td>\n",
       "      <td>GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...</td>\n",
       "      <td>-0.051746</td>\n",
       "      <td>0.144118</td>\n",
       "      <td>0.298042</td>\n",
       "      <td>-0.152253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.115361</td>\n",
       "      <td>0.142490</td>\n",
       "      <td>0.411190</td>\n",
       "      <td>-0.028281</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>304278 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Variant                                     Allele 1 Oligo  \\\n",
       "0           chr1:14677:G:A  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "1          chr1:522224:G:A  TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...   \n",
       "2          chr1:729679:C:G  TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...   \n",
       "3          chr1:741397:A:G  TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...   \n",
       "4          chr1:752566:G:A  AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...   \n",
       "...                    ...                                                ...   \n",
       "304273   chr4:57975328:A:G  TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...   \n",
       "304274  chr20:36661369:C:A  GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...   \n",
       "304275   chr8:60683160:C:T  GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...   \n",
       "304276   chr7:31046388:G:C  CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...   \n",
       "304277  chr10:10216373:A:G  GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...   \n",
       "\n",
       "                                           Allele 2 Oligo  \\\n",
       "0       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "1       TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...   \n",
       "2       TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...   \n",
       "3       TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...   \n",
       "4       AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...   \n",
       "...                                                   ...   \n",
       "304273  TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...   \n",
       "304274  GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...   \n",
       "304275  GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...   \n",
       "304276  CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...   \n",
       "304277  GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...   \n",
       "\n",
       "        allele1.log2FC_A549  allele1.log2FC_HEPG2  allele1.log2FC_K562  \\\n",
       "0                       NaN              0.667516            -0.704147   \n",
       "1                       NaN              0.627627             0.200635   \n",
       "2                       NaN              0.273272            -0.051526   \n",
       "3                       NaN             -0.114727             0.057421   \n",
       "4                       NaN             -0.067281            -0.140224   \n",
       "...                     ...                   ...                  ...   \n",
       "304273             0.764494              0.243506             0.507141   \n",
       "304274            -0.475731             -0.325048            -0.227184   \n",
       "304275            -0.080497             -0.020381            -0.070786   \n",
       "304276             0.359767              1.417800             2.603072   \n",
       "304277            -0.051746              0.144118             0.298042   \n",
       "\n",
       "        allele1.log2FC_SKNSH  allele1.log2FC_HCT116  allele2.log2FC_A549  \\\n",
       "0                   0.538641               0.159069                  NaN   \n",
       "1                  -0.088036               0.771700                  NaN   \n",
       "2                  -0.331275              -0.340771                  NaN   \n",
       "3                  -0.474388               0.402556                  NaN   \n",
       "4                   0.091286              -0.329839                  NaN   \n",
       "...                      ...                    ...                  ...   \n",
       "304273              1.145670                    NaN             0.539478   \n",
       "304274             -0.543515                    NaN            -0.350664   \n",
       "304275              0.394575                    NaN            -0.151279   \n",
       "304276              1.690492                    NaN             0.913629   \n",
       "304277             -0.152253                    NaN            -0.115361   \n",
       "\n",
       "        allele2.log2FC_HEPG2  allele2.log2FC_K562  allele2.log2FC_SKNSH  \\\n",
       "0                   0.415984             0.554099              0.441904   \n",
       "1                   0.316303            -0.015048             -0.231999   \n",
       "2                   0.062887            -0.699992             -0.469590   \n",
       "3                   0.040862            -0.073117              0.072672   \n",
       "4                  -0.149188            -0.090697              0.094279   \n",
       "...                      ...                  ...                   ...   \n",
       "304273              0.220710             0.458261              0.885200   \n",
       "304274              0.008168            -0.528934             -0.164957   \n",
       "304275              0.258262            -0.061669              0.296101   \n",
       "304276              2.021390             3.804445              2.135197   \n",
       "304277              0.142490             0.411190             -0.028281   \n",
       "\n",
       "        allele2.log2FC_HCT116  \n",
       "0                   -0.288019  \n",
       "1                    0.587129  \n",
       "2                   -0.604452  \n",
       "3                    0.264375  \n",
       "4                   -0.148007  \n",
       "...                       ...  \n",
       "304273                    NaN  \n",
       "304274                    NaN  \n",
       "304275                    NaN  \n",
       "304276                    NaN  \n",
       "304277                    NaN  \n",
       "\n",
       "[304278 rows x 13 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_df.to_csv('SirajMPRA_raw_data.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2021109/3571242790.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_allele_1['is_ref'] = True\n",
      "/tmp/ipykernel_2021109/3571242790.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_allele_2['is_ref'] = False\n"
     ]
    }
   ],
   "source": [
    "df_allele_1 = combined_df[['Variant', 'Allele 1 Oligo', 'allele1.log2FC_A549', 'allele1.log2FC_HEPG2', 'allele1.log2FC_K562', 'allele1.log2FC_SKNSH', 'allele1.log2FC_HCT116']]\n",
    "df_allele_2 = combined_df[['Variant', 'Allele 2 Oligo', 'allele2.log2FC_A549', 'allele2.log2FC_HEPG2', 'allele2.log2FC_K562', 'allele2.log2FC_SKNSH', 'allele2.log2FC_HCT116']]\n",
    "df_allele_1['is_ref'] = True\n",
    "df_allele_2['is_ref'] = False\n",
    "\n",
    "df_allele_1 = df_allele_1.rename(columns={'Variant': 'description', 'Allele 1 Oligo': 'seq', 'allele1.log2FC_A549': 'A549', 'allele1.log2FC_HEPG2': 'HepG2', 'allele1.log2FC_K562': 'K562', 'allele1.log2FC_SKNSH': 'SKNSH', 'allele1.log2FC_HCT116': 'HCT116'})\n",
    "df_allele_2 = df_allele_2.rename(columns={'Variant': 'description', 'Allele 2 Oligo': 'seq', 'allele2.log2FC_A549': 'A549', 'allele2.log2FC_HEPG2': 'HepG2', 'allele2.log2FC_K562': 'K562', 'allele2.log2FC_SKNSH': 'SKNSH', 'allele2.log2FC_HCT116': 'HCT116'})\n",
    "\n",
    "concat_df = pd.concat([df_allele_1, df_allele_2], axis=0)\n",
    "concat_df = concat_df.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>seq</th>\n",
       "      <th>A549</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>SKNSH</th>\n",
       "      <th>HCT116</th>\n",
       "      <th>is_ref</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:14677:G:A</td>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.667516</td>\n",
       "      <td>-0.704147</td>\n",
       "      <td>0.538641</td>\n",
       "      <td>0.159069</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:522224:G:A</td>\n",
       "      <td>TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.627627</td>\n",
       "      <td>0.200635</td>\n",
       "      <td>-0.088036</td>\n",
       "      <td>0.771700</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:729679:C:G</td>\n",
       "      <td>TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.273272</td>\n",
       "      <td>-0.051526</td>\n",
       "      <td>-0.331275</td>\n",
       "      <td>-0.340771</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:741397:A:G</td>\n",
       "      <td>TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.114727</td>\n",
       "      <td>0.057421</td>\n",
       "      <td>-0.474388</td>\n",
       "      <td>0.402556</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:752566:G:A</td>\n",
       "      <td>AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.067281</td>\n",
       "      <td>-0.140224</td>\n",
       "      <td>0.091286</td>\n",
       "      <td>-0.329839</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608551</th>\n",
       "      <td>chr4:57975328:A:G</td>\n",
       "      <td>TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...</td>\n",
       "      <td>0.539478</td>\n",
       "      <td>0.220710</td>\n",
       "      <td>0.458261</td>\n",
       "      <td>0.885200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608552</th>\n",
       "      <td>chr20:36661369:C:A</td>\n",
       "      <td>GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...</td>\n",
       "      <td>-0.350664</td>\n",
       "      <td>0.008168</td>\n",
       "      <td>-0.528934</td>\n",
       "      <td>-0.164957</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608553</th>\n",
       "      <td>chr8:60683160:C:T</td>\n",
       "      <td>GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...</td>\n",
       "      <td>-0.151279</td>\n",
       "      <td>0.258262</td>\n",
       "      <td>-0.061669</td>\n",
       "      <td>0.296101</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608554</th>\n",
       "      <td>chr7:31046388:G:C</td>\n",
       "      <td>CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...</td>\n",
       "      <td>0.913629</td>\n",
       "      <td>2.021390</td>\n",
       "      <td>3.804445</td>\n",
       "      <td>2.135197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608555</th>\n",
       "      <td>chr10:10216373:A:G</td>\n",
       "      <td>GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...</td>\n",
       "      <td>-0.115361</td>\n",
       "      <td>0.142490</td>\n",
       "      <td>0.411190</td>\n",
       "      <td>-0.028281</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>608556 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               description                                                seq  \\\n",
       "0           chr1:14677:G:A  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   \n",
       "1          chr1:522224:G:A  TCCTGTGGTTCTCTGTTCTTTTCTCAGGTGTGTCCTTAACCTTGCA...   \n",
       "2          chr1:729679:C:G  TCCATGCAGTGATGGTCACCTTCTCTAACAGGCATTTCAGAGTGAG...   \n",
       "3          chr1:741397:A:G  TTAGTCAATTAGCCAAAAAGAATTGGCAGTTAATGTCAATTAATTG...   \n",
       "4          chr1:752566:G:A  AAAGAATTTTTGGCTAAGTTCCCAAAAACGATTGCAACAAAACAAA...   \n",
       "...                    ...                                                ...   \n",
       "608551   chr4:57975328:A:G  TTTGCCCACAAATAATTAGCTAGTTAGATTACTGTTTGAGAACCTA...   \n",
       "608552  chr20:36661369:C:A  GGTCAACAGACACATTGGGAAGGCTAAGTTAGTTCCCTCCACCTTT...   \n",
       "608553   chr8:60683160:C:T  GGCCATGGAGTTTATTCAAATTGTATATCTAAACAATTTTAGTATT...   \n",
       "608554   chr7:31046388:G:C  CTCAGGGCTCCATCCCCGCCTGGCCCCATCACCCCAGGCCCTCTGG...   \n",
       "608555  chr10:10216373:A:G  GAATTTATTGTAAAAAGAGAACATGCATTAACATCAGGCACAATAA...   \n",
       "\n",
       "            A549     HepG2      K562     SKNSH    HCT116  is_ref  \n",
       "0            NaN  0.667516 -0.704147  0.538641  0.159069    True  \n",
       "1            NaN  0.627627  0.200635 -0.088036  0.771700    True  \n",
       "2            NaN  0.273272 -0.051526 -0.331275 -0.340771    True  \n",
       "3            NaN -0.114727  0.057421 -0.474388  0.402556    True  \n",
       "4            NaN -0.067281 -0.140224  0.091286 -0.329839    True  \n",
       "...          ...       ...       ...       ...       ...     ...  \n",
       "608551  0.539478  0.220710  0.458261  0.885200       NaN   False  \n",
       "608552 -0.350664  0.008168 -0.528934 -0.164957       NaN   False  \n",
       "608553 -0.151279  0.258262 -0.061669  0.296101       NaN   False  \n",
       "608554  0.913629  2.021390  3.804445  2.135197       NaN   False  \n",
       "608555 -0.115361  0.142490  0.411190 -0.028281       NaN   False  \n",
       "\n",
       "[608556 rows x 8 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concat_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_df[['chr', 'pos', 'ref', 'alt']] = concat_df['description'].str.split(':', expand=True)\n",
    "concat_df = concat_df.drop(columns=['description'])\n",
    "concat_df['pos'] = concat_df['pos'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_df = concat_df.sort_values(by=['chr', 'pos'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq</th>\n",
       "      <th>A549</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>SKNSH</th>\n",
       "      <th>HCT116</th>\n",
       "      <th>is_ref</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.667516</td>\n",
       "      <td>-0.704147</td>\n",
       "      <td>0.538641</td>\n",
       "      <td>0.159069</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304278</th>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.415984</td>\n",
       "      <td>0.554099</td>\n",
       "      <td>0.441904</td>\n",
       "      <td>-0.288019</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126810</th>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.269937</td>\n",
       "      <td>-0.391222</td>\n",
       "      <td>0.752052</td>\n",
       "      <td>1.100437</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431088</th>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.151995</td>\n",
       "      <td>-0.659023</td>\n",
       "      <td>0.204721</td>\n",
       "      <td>1.050619</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116966</th>\n",
       "      <td>AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.211076</td>\n",
       "      <td>-0.007174</td>\n",
       "      <td>0.499376</td>\n",
       "      <td>0.257572</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>64764</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421242</th>\n",
       "      <td>ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.615976</td>\n",
       "      <td>1.927090</td>\n",
       "      <td>0.798263</td>\n",
       "      <td>1.035440</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233098</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153505</th>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.409758</td>\n",
       "      <td>0.556338</td>\n",
       "      <td>2.843295</td>\n",
       "      <td>1.495515</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457783</th>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.894634</td>\n",
       "      <td>-0.072647</td>\n",
       "      <td>1.973661</td>\n",
       "      <td>0.794848</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116965</th>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312172</td>\n",
       "      <td>0.165634</td>\n",
       "      <td>-0.041383</td>\n",
       "      <td>0.444472</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421243</th>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.370818</td>\n",
       "      <td>0.009694</td>\n",
       "      <td>-0.035569</td>\n",
       "      <td>0.407723</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>608556 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      seq  A549     HepG2  \\\n",
       "0       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   NaN  0.667516   \n",
       "304278  AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   NaN  0.415984   \n",
       "126810  CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   NaN  1.269937   \n",
       "431088  CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   NaN  1.151995   \n",
       "116966  AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...   NaN -0.211076   \n",
       "...                                                   ...   ...       ...   \n",
       "421242  ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...   NaN  0.615976   \n",
       "153505  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   NaN  1.409758   \n",
       "457783  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   NaN  0.894634   \n",
       "116965  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   NaN  0.312172   \n",
       "421243  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   NaN  0.370818   \n",
       "\n",
       "            K562     SKNSH    HCT116  is_ref   chr        pos ref alt  \n",
       "0      -0.704147  0.538641  0.159069    True  chr1      14677   G   A  \n",
       "304278  0.554099  0.441904 -0.288019   False  chr1      14677   G   A  \n",
       "126810 -0.391222  0.752052  1.100437    True  chr1      63697   T   C  \n",
       "431088 -0.659023  0.204721  1.050619   False  chr1      63697   T   C  \n",
       "116966 -0.007174  0.499376  0.257572    True  chr1      64764   C   T  \n",
       "...          ...       ...       ...     ...   ...        ...  ..  ..  \n",
       "421242  1.927090  0.798263  1.035440   False  chrX  155233098   T   C  \n",
       "153505  0.556338  2.843295  1.495515    True  chrX  155233794   A   G  \n",
       "457783 -0.072647  1.973661  0.794848   False  chrX  155233794   A   G  \n",
       "116965  0.165634 -0.041383  0.444472    True  chrX  155234730   T   C  \n",
       "421243  0.009694 -0.035569  0.407723   False  chrX  155234730   T   C  \n",
       "\n",
       "[608556 rows x 11 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concat_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_df.to_csv(\"SirajMPRA_data_concat.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq</th>\n",
       "      <th>A549</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>SKNSH</th>\n",
       "      <th>HCT116</th>\n",
       "      <th>is_ref</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.667516</td>\n",
       "      <td>-0.704147</td>\n",
       "      <td>0.538641</td>\n",
       "      <td>0.159069</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.415984</td>\n",
       "      <td>0.554099</td>\n",
       "      <td>0.441904</td>\n",
       "      <td>-0.288019</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.269937</td>\n",
       "      <td>-0.391222</td>\n",
       "      <td>0.752052</td>\n",
       "      <td>1.100437</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.151995</td>\n",
       "      <td>-0.659023</td>\n",
       "      <td>0.204721</td>\n",
       "      <td>1.050619</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.211076</td>\n",
       "      <td>-0.007174</td>\n",
       "      <td>0.499376</td>\n",
       "      <td>0.257572</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>64764</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608551</th>\n",
       "      <td>ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.615976</td>\n",
       "      <td>1.927090</td>\n",
       "      <td>0.798263</td>\n",
       "      <td>1.035440</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233098</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608552</th>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.409758</td>\n",
       "      <td>0.556338</td>\n",
       "      <td>2.843295</td>\n",
       "      <td>1.495515</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608553</th>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.894634</td>\n",
       "      <td>-0.072647</td>\n",
       "      <td>1.973661</td>\n",
       "      <td>0.794848</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608554</th>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312172</td>\n",
       "      <td>0.165634</td>\n",
       "      <td>-0.041383</td>\n",
       "      <td>0.444472</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608555</th>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.370818</td>\n",
       "      <td>0.009694</td>\n",
       "      <td>-0.035569</td>\n",
       "      <td>0.407723</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>608556 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      seq  A549     HepG2  \\\n",
       "0       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   NaN  0.667516   \n",
       "1       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   NaN  0.415984   \n",
       "2       CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   NaN  1.269937   \n",
       "3       CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   NaN  1.151995   \n",
       "4       AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...   NaN -0.211076   \n",
       "...                                                   ...   ...       ...   \n",
       "608551  ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...   NaN  0.615976   \n",
       "608552  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   NaN  1.409758   \n",
       "608553  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   NaN  0.894634   \n",
       "608554  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   NaN  0.312172   \n",
       "608555  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   NaN  0.370818   \n",
       "\n",
       "            K562     SKNSH    HCT116  is_ref   chr        pos ref alt  \n",
       "0      -0.704147  0.538641  0.159069    True  chr1      14677   G   A  \n",
       "1       0.554099  0.441904 -0.288019   False  chr1      14677   G   A  \n",
       "2      -0.391222  0.752052  1.100437    True  chr1      63697   T   C  \n",
       "3      -0.659023  0.204721  1.050619   False  chr1      63697   T   C  \n",
       "4      -0.007174  0.499376  0.257572    True  chr1      64764   C   T  \n",
       "...          ...       ...       ...     ...   ...        ...  ..  ..  \n",
       "608551  1.927090  0.798263  1.035440   False  chrX  155233098   T   C  \n",
       "608552  0.556338  2.843295  1.495515    True  chrX  155233794   A   G  \n",
       "608553 -0.072647  1.973661  0.794848   False  chrX  155233794   A   G  \n",
       "608554  0.165634 -0.041383  0.444472    True  chrX  155234730   T   C  \n",
       "608555  0.009694 -0.035569  0.407723   False  chrX  155234730   T   C  \n",
       "\n",
       "[608556 rows x 11 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concat_df = pd.read_csv(\"SirajMPRA_data_concat.csv\")\n",
    "concat_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "seq            0\n",
      "A549      356964\n",
      "HepG2       2026\n",
      "K562        2704\n",
      "SKNSH       2888\n",
      "HCT116    238340\n",
      "is_ref         0\n",
      "chr            0\n",
      "pos            0\n",
      "ref            0\n",
      "alt            0\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(pd.isna(concat_df).sum(0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  5%|▍         | 15103/304278 [00:00<00:07, 38331.86it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 8151\n",
      "Cannot convert line 8198\n",
      "Cannot convert line 8212\n",
      "Cannot convert line 8359\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 13%|█▎        | 40137/304278 [00:01<00:06, 40865.94it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 32334\n",
      "Cannot convert line 32345\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 20%|██        | 61393/304278 [00:01<00:06, 39565.94it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 55226\n",
      "Cannot convert line 55249\n",
      "Cannot convert line 55291\n",
      "Cannot convert line 55299\n",
      "Cannot convert line 55312\n",
      "Cannot convert line 55840\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 33%|███▎      | 99617/304278 [00:02<00:05, 39598.40it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 90946\n",
      "Cannot convert line 91033\n",
      "Cannot convert line 92230\n",
      "Cannot convert line 96730\n",
      "Cannot convert line 96732\n",
      "Cannot convert line 97867\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 35%|███▌      | 107882/304278 [00:02<00:04, 40431.90it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 100779\n",
      "Cannot convert line 100780\n",
      "Cannot convert line 100781\n",
      "Cannot convert line 100787\n",
      "Cannot convert line 107324\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 40%|████      | 122981/304278 [00:03<00:06, 28472.98it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 116388\n",
      "Cannot convert line 116414\n",
      "Cannot convert line 116481\n",
      "Cannot convert line 120369\n",
      "Cannot convert line 120459\n",
      "Cannot convert line 120547\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 43%|████▎     | 130897/304278 [00:03<00:05, 32917.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 126416\n",
      "Cannot convert line 126765\n",
      "Cannot convert line 126834\n",
      "Cannot convert line 126889\n",
      "Cannot convert line 126893\n",
      "Cannot convert line 127164\n",
      "Cannot convert line 127254\n",
      "Cannot convert line 127259\n",
      "Cannot convert line 127263\n",
      "Cannot convert line 127361\n",
      "Cannot convert line 127395\n",
      "Cannot convert line 127531\n",
      "Cannot convert line 127535\n",
      "Cannot convert line 127569\n",
      "Cannot convert line 127684\n",
      "Cannot convert line 127710\n",
      "Cannot convert line 129582\n",
      "Cannot convert line 129583\n",
      "Cannot convert line 129585\n",
      "Cannot convert line 129594\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 47%|████▋     | 142746/304278 [00:03<00:04, 36444.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 136145\n",
      "Cannot convert line 136250\n",
      "Cannot convert line 136251\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 53%|█████▎    | 159771/304278 [00:04<00:03, 40032.82it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 154910\n",
      "Cannot convert line 155151\n",
      "Cannot convert line 155154\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 57%|█████▋    | 172155/304278 [00:04<00:03, 39723.96it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 165160\n",
      "Cannot convert line 165261\n",
      "Cannot convert line 165325\n",
      "Cannot convert line 166012\n",
      "Cannot convert line 169496\n",
      "Cannot convert line 169520\n",
      "Cannot convert line 170549\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 66%|██████▋   | 201691/304278 [00:05<00:02, 38636.88it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 194180\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 72%|███████▏  | 217942/304278 [00:05<00:02, 38652.47it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 211182\n",
      "Cannot convert line 215657\n",
      "Cannot convert line 216604\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 88%|████████▊ | 267204/304278 [00:07<00:00, 39549.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 260152\n",
      "Cannot convert line 260199\n",
      "Cannot convert line 261403\n",
      "Cannot convert line 261441\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 91%|█████████ | 275570/304278 [00:07<00:00, 40304.54it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 269859\n",
      "Cannot convert line 270544\n",
      "Cannot convert line 270549\n",
      "Cannot convert line 270555\n",
      "Cannot convert line 271046\n",
      "Cannot convert line 271078\n",
      "Cannot convert line 271109\n",
      "Cannot convert line 271118\n",
      "Cannot convert line 271167\n",
      "Cannot convert line 271227\n",
      "Cannot convert line 271254\n",
      "Cannot convert line 271278\n",
      "Cannot convert line 271323\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 95%|█████████▍| 287878/304278 [00:07<00:00, 39963.42it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 279685\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 96%|█████████▌| 291992/304278 [00:07<00:00, 40307.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 291967\n",
      "Cannot convert line 291968\n",
      "Cannot convert line 291969\n",
      "Cannot convert line 291986\n",
      "Cannot convert line 291988\n",
      "Cannot convert line 291996\n",
      "Cannot convert line 292016\n",
      "Cannot convert line 292026\n",
      "Cannot convert line 292035\n",
      "Cannot convert line 292046\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 99%|█████████▊| 299773/304278 [00:08<00:00, 26101.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 292687\n",
      "Cannot convert line 300822\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 304278/304278 [00:08<00:00, 37123.22it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cannot convert line 301692\n",
      "Cannot convert line 301757\n",
      "Cannot convert line 301830\n",
      "Cannot convert line 302027\n",
      "Cannot convert line 302360\n",
      "Cannot convert line 302462\n",
      "Cannot convert line 303616\n"
     ]
    }
   ],
   "source": [
    "# from pyliftover import LiftOver\n",
    "# lo = LiftOver('hg19', 'hg38')\n",
    "# # lo.convert_coordinate('chr1', 1000000)\n",
    "\n",
    "# new_df = []\n",
    "# for i in tqdm(range(len(df))):\n",
    "#     info = df.iloc[i, 0]\n",
    "#     chr, pos, base1, base2 = info.split(':')\n",
    "#     pos = int(pos)\n",
    "\n",
    "#     converted = lo.convert_coordinate(chr, pos)\n",
    "#     if converted and len(converted) > 0:\n",
    "#         hg38_chr, hg38_pos = converted[0][0], converted[0][1]\n",
    "#     else:\n",
    "#         hg38_chr, hg38_pos = 'NaN', 0\n",
    "#         print(f'Cannot convert line {i}')\n",
    "    \n",
    "#     new_df.append([chr, pos, hg38_chr, hg38_pos, base1, base2])\n",
    "\n",
    "# new_df = pd.DataFrame(new_df, columns=['hg19_chr','hg19_pos','hg38_chr','hg38_pos','base1','base2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = pd.read_csv('SirajMPRA_data_concat.csv')\n",
    "d = d[d['seq'].str.len() == 200]\n",
    "d.to_csv('SirajMPRA_len200.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = pd.read_csv('SirajMPRA_len200.csv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq</th>\n",
       "      <th>A549</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>SKNSH</th>\n",
       "      <th>HCT116</th>\n",
       "      <th>is_ref</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.667516</td>\n",
       "      <td>-0.704147</td>\n",
       "      <td>0.538641</td>\n",
       "      <td>0.159069</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.415984</td>\n",
       "      <td>0.554099</td>\n",
       "      <td>0.441904</td>\n",
       "      <td>-0.288019</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14677</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.269937</td>\n",
       "      <td>-0.391222</td>\n",
       "      <td>0.752052</td>\n",
       "      <td>1.100437</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.151995</td>\n",
       "      <td>-0.659023</td>\n",
       "      <td>0.204721</td>\n",
       "      <td>1.050619</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63697</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.211076</td>\n",
       "      <td>-0.007174</td>\n",
       "      <td>0.499376</td>\n",
       "      <td>0.257572</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1</td>\n",
       "      <td>64764</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585600</th>\n",
       "      <td>ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.615976</td>\n",
       "      <td>1.927090</td>\n",
       "      <td>0.798263</td>\n",
       "      <td>1.035440</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233098</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585601</th>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.409758</td>\n",
       "      <td>0.556338</td>\n",
       "      <td>2.843295</td>\n",
       "      <td>1.495515</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585602</th>\n",
       "      <td>TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.894634</td>\n",
       "      <td>-0.072647</td>\n",
       "      <td>1.973661</td>\n",
       "      <td>0.794848</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155233794</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585603</th>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312172</td>\n",
       "      <td>0.165634</td>\n",
       "      <td>-0.041383</td>\n",
       "      <td>0.444472</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585604</th>\n",
       "      <td>TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.370818</td>\n",
       "      <td>0.009694</td>\n",
       "      <td>-0.035569</td>\n",
       "      <td>0.407723</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155234730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>585605 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      seq  A549     HepG2  \\\n",
       "0       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   NaN  0.667516   \n",
       "1       AGTGCTGGTTCCGTCACCCCCTCCCAAGGAAGTAGGTCTGAGCAGC...   NaN  0.415984   \n",
       "2       CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   NaN  1.269937   \n",
       "3       CATACATTTTTATTCTGGTCACTGTCCAACGACATTCCTCAAATGA...   NaN  1.151995   \n",
       "4       AATAGACTCTAGTACTCTGTATTATGCAAAATTTGTCTATGTTACA...   NaN -0.211076   \n",
       "...                                                   ...   ...       ...   \n",
       "585600  ATCATTTACTGGTGACTGCCCTGCTAGGGTGTCAGCTGTCAGATCC...   NaN  0.615976   \n",
       "585601  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   NaN  1.409758   \n",
       "585602  TGGTTACTGCAGGGGCAGGGTTTTGGCAGGAAATAAACATGCACGG...   NaN  0.894634   \n",
       "585603  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   NaN  0.312172   \n",
       "585604  TGGTGAGTTGCTTCTGTGCACACACTTTTGTTTATGAGTGTGCATG...   NaN  0.370818   \n",
       "\n",
       "            K562     SKNSH    HCT116  is_ref   chr        pos ref alt  \n",
       "0      -0.704147  0.538641  0.159069    True  chr1      14677   G   A  \n",
       "1       0.554099  0.441904 -0.288019   False  chr1      14677   G   A  \n",
       "2      -0.391222  0.752052  1.100437    True  chr1      63697   T   C  \n",
       "3      -0.659023  0.204721  1.050619   False  chr1      63697   T   C  \n",
       "4      -0.007174  0.499376  0.257572    True  chr1      64764   C   T  \n",
       "...          ...       ...       ...     ...   ...        ...  ..  ..  \n",
       "585600  1.927090  0.798263  1.035440   False  chrX  155233098   T   C  \n",
       "585601  0.556338  2.843295  1.495515    True  chrX  155233794   A   G  \n",
       "585602 -0.072647  1.973661  0.794848   False  chrX  155233794   A   G  \n",
       "585603  0.165634 -0.041383  0.444472    True  chrX  155234730   T   C  \n",
       "585604  0.009694 -0.035569  0.407723   False  chrX  155234730   T   C  \n",
       "\n",
       "[585605 rows x 11 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "# from pyfaidx import Fasta\n",
    "# genome = Fasta('/home/hxcai/genome/hg19.fa')\n",
    "\n",
    "# seqs = []\n",
    "# for i in tqdm(range(len(d))):\n",
    "#     chr, start, end = d.iloc[i][['chr','start', 'end']]\n",
    "#     seq = genome[chr][start: end].seq.upper()\n",
    "#     seqs.append(seq)\n",
    "\n",
    "# d['seq'] = seqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1:761957:A:AT\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTCATAAGCTGAACCTGACACAATAAATTTAAAAGGGAAACTAATTTGGAAATCAGAAAACCACTAAGGAATTTGGGAATTAGGCTTCTGCTGCCCTCTCTGCTACTGACGGTCAAGGCCTCCTCATTGTATTCTGTCCTCCATATCTCTGCTGATTCCCATT\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTCATAAGCTGAACCTGACACAATAAATTTAAAAGGGAAACTAATTTGGAAATCAGAAAACCATCTAAGGAATTTGGGAATTAGGCTTCTGCTGCCCTCTCTGCTACTGACGGTCAAGGCCTCCTCATTGTATTCTGTCCTCCATATCTCTGCTGATTCCCATT\n",
      "TTTGTCCTCAGTGGGGGCATCGGATGCACGGGCACCATCATCGTGATTGACATCCTGGTGGACGTCATCCGCAGGCAGGGCGAGCCGCTCCACTCTGGGTCCCCCCGCCCTGCCCTGCTATAGCCCCACCCCTCCGGGCGACCCCACCCCTCCGGACGACCCCACCCCTCAGGACGACCCCACCCCTCAGGACGACCCCA\n",
      "199 200 200\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTCATAAGCTGAACCTGACACAATAAATTTAAAAGGGAAACTAATTTGGAAATCAGAAAACCA-CTAAGGAATTTGGGAATTAGGCTTCTGCTGCCCTCTCTGCTACTGACGGTCAAGGCCTCCTCATTGTATTCTGTCCTCCATATCTCTGCTGATTCCCATT\n",
      "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTCATAAGCTGAACCTGACACAATAAATTTAAAAGGGAAACTAATTTGGAAATCAGAAAACCATCTAAGGAATTTGGGAATTAGGCTTCTGCTGCCCTCTCTGCTACTGACGGTCAAGGCCTCCTCATTGTATTCTGTCCTCCATATCTCTGCTGATTCCCATT\n",
      "  Score=397.5\n",
      "\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTC--ATAAGCTGAACCTGACACAATAAATT-TAAAAGGGAAACTAAT--T--TGGAAATCAGAAAACCACTAAGGAATTTGGGAATTAG--GCT----TCTG-------CTGCCCT-CTCTGCTACT-G----A------CGGTC-A----A-------GG-------C-CTCCTCATTGTATTCTGTCCTCCATATCTCTG--CTGATTCCCATT\n",
      "||    |.|.|   |.||  ||    |    | |||.||  ||  ||   | |.|.|||.|| .||. |.|..|    ||  ||  |  ||||..||    |.||.|  |||.|   |||  ..||  |||    ||||       |.||||| |.|||||| | |    |      |||.| |    |       ||       | |.|||||  |.|  | |.|| |||...|||.|  | || .||||  \n",
      "TT----TGTCC---TCAG--TG----G----G-GGCATCGGAT--GC---A-CGGGCACCAT-CATCGTGATTG----AC--ATCCTGGTGGACGTC----ATCCGC--AGGCA---GGG--CGAGCCGCTCCACTCTGGGTCCCCCCGCCCTGCCCTGCTA-TAGCCCCACCCCTCCGGGCGACCCCACCCCTCCGGACGACCCCACCCCTCA--GGA--C-GACC-CCACCCCTCAGGAC-GA-CCCCA--\n",
      "  Score=204.9\n",
      "\n",
      "TTTCAATATACAATTAAGACTGAAAAGTATTGAGGCCTC--ATAAGCTGAACCTGACACAATAAATT-TAAAAGGGAAACTAAT--T--TGGAAATCAGAAAACCATCTAAGGAATTTGGGAATTAG--GCT----TCTG-------CTGCCCT-CTCTGCTACT-G----A------CGGTC-A----A-------GG-------C-CTCCTCATTGTATTCTGTCCTCCATATCTCTG--CTGATTCCCATT\n",
      "||    |.|.|   |.||  ||    |    | |||.||  ||  ||   | |.|.|||.|| .||. |.|..|    ||  ||  |  ||||..||    |.|| .|  |||.|   |||  ..||  |||    ||||       |.||||| |.|||||| | |    |      |||.| |    |       ||       | |.|||||  |.|  | |.|| |||...|||.|  | || .||||  \n",
      "TT----TGTCC---TCAG--TG----G----G-GGCATCGGAT--GC---A-CGGGCACCAT-CATCGTGATTG----AC--ATCCTGGTGGACGTC----ATCC-GC--AGGCA---GGG--CGAGCCGCTCCACTCTGGGTCCCCCCGCCCTGCCCTGCTA-TAGCCCCACCCCTCCGGGCGACCCCACCCCTCCGGACGACCCCACCCCTCA--GGA--C-GACC-CCACCCCTCAGGAC-GA-CCCCA--\n",
      "  Score=204.4\n",
      "\n"
     ]
    }
   ],
   "source": [
    "info, seq1, seq2 = df.iloc[15]\n",
    "\n",
    "print(info)\n",
    "print(seq1)\n",
    "print(seq2)\n",
    "\n",
    "from pyfaidx import Fasta\n",
    "genome = Fasta('/home/hxcai/genome/hg19.fa')\n",
    "seq3 = genome['chr1'][1004389-100: 1004389+100].seq.upper()\n",
    "\n",
    "print(seq3)\n",
    "print(len(seq1), len(seq2), len(seq3))\n",
    "\n",
    "alignments = align.globalms(seq1, seq2, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "\n",
    "alignments = align.globalms(seq1, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "\n",
    "alignments = align.globalms(seq2, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1:759444:ATATT:A\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "200 196 200\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||    ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAA----TATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "  Score=391.2\n",
      "\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "  Score=400\n",
      "\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAA----TATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||    ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "GTGAGTATACAGTCAAAATAGTTGTGAAAAGAAAAAAATAAAGTTGTCAAATTTCATGTAACCTATGTTATTTGGAAATTTAAAATAAGTTATATAAATATTTATAACAATGTTAAACAATTTCATAAGTGGACTGAATAAAACATAATATTTTATCATATACAATAGCTCCTGCCTTCCTGAGGAAAACATAAAATCAA\n",
      "  Score=391.2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "info, seq1, seq2 = df.iloc[10]\n",
    "\n",
    "print(info)\n",
    "print(seq1)\n",
    "print(seq2)\n",
    "\n",
    "from pyfaidx import Fasta\n",
    "genome = Fasta('/home/hxcai/genome/hg19.fa')\n",
    "seq3 = genome['chr1'][759444-100+2: 759444+100+2].seq.upper()\n",
    "\n",
    "print(seq3)\n",
    "print(len(seq1), len(seq2), len(seq3))\n",
    "\n",
    "alignments = align.globalms(seq1, seq2, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "\n",
    "alignments = align.globalms(seq1, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))\n",
    "\n",
    "alignments = align.globalms(seq2, seq3, 2, 0, -.5, -.1)\n",
    "print(format_alignment(*alignments[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "torch",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "vscode": {
   "interpreter": {
    "hash": "8e62da015c2b8e1460a389c60847c46fcac689812463b880d1463e5f1a16b0d6"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
