{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.path.append('..')\n",
    "from MPRA_predict.utils import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>tss_distance</th>\n",
       "      <th>ma_samples</th>\n",
       "      <th>ma_count</th>\n",
       "      <th>maf</th>\n",
       "      <th>pval_nominal</th>\n",
       "      <th>slope</th>\n",
       "      <th>slope_se</th>\n",
       "      <th>pval_nominal_threshold</th>\n",
       "      <th>min_pval_nominal</th>\n",
       "      <th>pval_beta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_64764_C_T_b38</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>35211</td>\n",
       "      <td>79</td>\n",
       "      <td>80</td>\n",
       "      <td>0.060</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.517</td>\n",
       "      <td>0.084</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_665098_G_A_b38</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>635545</td>\n",
       "      <td>140</td>\n",
       "      <td>146</td>\n",
       "      <td>0.109</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.463</td>\n",
       "      <td>0.063</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_666028_G_A_b38</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>636475</td>\n",
       "      <td>125</td>\n",
       "      <td>131</td>\n",
       "      <td>0.098</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.490</td>\n",
       "      <td>0.066</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_17556_C_T_b38</td>\n",
       "      <td>ENSG00000238009.6</td>\n",
       "      <td>-111667</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.761</td>\n",
       "      <td>0.156</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_58814_G_A_b38</td>\n",
       "      <td>ENSG00000238009.6</td>\n",
       "      <td>-70409</td>\n",
       "      <td>109</td>\n",
       "      <td>119</td>\n",
       "      <td>0.089</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.637</td>\n",
       "      <td>0.070</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2414648</th>\n",
       "      <td>chrX_155999818_C_T_b38</td>\n",
       "      <td>ENSG00000182484.15</td>\n",
       "      <td>-21008</td>\n",
       "      <td>419</td>\n",
       "      <td>515</td>\n",
       "      <td>0.384</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.182</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2414649</th>\n",
       "      <td>chrX_155999957_G_C_b38</td>\n",
       "      <td>ENSG00000182484.15</td>\n",
       "      <td>-20869</td>\n",
       "      <td>403</td>\n",
       "      <td>487</td>\n",
       "      <td>0.363</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.182</td>\n",
       "      <td>0.048</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2414650</th>\n",
       "      <td>chrX_156000131_G_A_b38</td>\n",
       "      <td>ENSG00000182484.15</td>\n",
       "      <td>-20695</td>\n",
       "      <td>402</td>\n",
       "      <td>485</td>\n",
       "      <td>0.362</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.181</td>\n",
       "      <td>0.048</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2414651</th>\n",
       "      <td>chrX_156001267_A_G_b38</td>\n",
       "      <td>ENSG00000182484.15</td>\n",
       "      <td>-19559</td>\n",
       "      <td>400</td>\n",
       "      <td>481</td>\n",
       "      <td>0.359</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.176</td>\n",
       "      <td>0.048</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2414652</th>\n",
       "      <td>chrX_156003433_T_C_b38</td>\n",
       "      <td>ENSG00000182484.15</td>\n",
       "      <td>-17393</td>\n",
       "      <td>421</td>\n",
       "      <td>532</td>\n",
       "      <td>0.397</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.182</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.050</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2414653 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     variant_id             gene_id  tss_distance  ma_samples  ma_count   maf  pval_nominal  slope  slope_se  pval_nominal_threshold  min_pval_nominal  pval_beta\n",
       "0            chr1_64764_C_T_b38   ENSG00000227232.5         35211          79        80 0.060         0.000  0.517     0.084                   0.000             0.000      0.000\n",
       "1           chr1_665098_G_A_b38   ENSG00000227232.5        635545         140       146 0.109         0.000  0.463     0.063                   0.000             0.000      0.000\n",
       "2           chr1_666028_G_A_b38   ENSG00000227232.5        636475         125       131 0.098         0.000  0.490     0.066                   0.000             0.000      0.000\n",
       "3            chr1_17556_C_T_b38   ENSG00000238009.6       -111667          27        27 0.020         0.000  0.761     0.156                   0.000             0.000      0.000\n",
       "4            chr1_58814_G_A_b38   ENSG00000238009.6        -70409         109       119 0.089         0.000  0.637     0.070                   0.000             0.000      0.000\n",
       "...                         ...                 ...           ...         ...       ...   ...           ...    ...       ...                     ...               ...        ...\n",
       "2414648  chrX_155999818_C_T_b38  ENSG00000182484.15        -21008         419       515 0.384         0.000  0.182     0.049                   0.001             0.000      0.050\n",
       "2414649  chrX_155999957_G_C_b38  ENSG00000182484.15        -20869         403       487 0.363         0.000  0.182     0.048                   0.001             0.000      0.050\n",
       "2414650  chrX_156000131_G_A_b38  ENSG00000182484.15        -20695         402       485 0.362         0.000  0.181     0.048                   0.001             0.000      0.050\n",
       "2414651  chrX_156001267_A_G_b38  ENSG00000182484.15        -19559         400       481 0.359         0.000  0.176     0.048                   0.001             0.000      0.050\n",
       "2414652  chrX_156003433_T_C_b38  ENSG00000182484.15        -17393         421       532 0.397         0.000  0.182     0.049                   0.001             0.000      0.050\n",
       "\n",
       "[2414653 rows x 12 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_name = '../data/GTEx_eQTL/GTEx_Analysis_v8_eQTL/Whole_Blood.v8.signif_variant_gene_pairs.txt.gz'\n",
    "df = pd.read_csv(file_name, sep='\\t')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>916119</td>\n",
       "      <td>chr1_916119_A_G_b38</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1203822</td>\n",
       "      <td>chr1_1203822_T_C_b38</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1680672</td>\n",
       "      <td>chr1_1680672_C_T_b38</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2174439</td>\n",
       "      <td>chr1_2174439_G_A_b38</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2177928</td>\n",
       "      <td>chr1_2177928_G_A_b38</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1891</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155528816</td>\n",
       "      <td>chrX_155528816_T_C_b38</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1892</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155649454</td>\n",
       "      <td>chrX_155649454_A_G_b38</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1893</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155668742</td>\n",
       "      <td>chrX_155668742_C_A_b38</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1894</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155679088</td>\n",
       "      <td>chrX_155679088_G_C_b38</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1895</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155687184</td>\n",
       "      <td>chrX_155687184_G_T_b38</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1896 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr        pos              variant_id ref alt  1  2   3\n",
       "0     chr1     916119     chr1_916119_A_G_b38   G   A  .  . NaN\n",
       "1     chr1    1203822    chr1_1203822_T_C_b38   T   C  .  . NaN\n",
       "2     chr1    1680672    chr1_1680672_C_T_b38   T   C  .  . NaN\n",
       "3     chr1    2174439    chr1_2174439_G_A_b38   G   A  .  . NaN\n",
       "4     chr1    2177928    chr1_2177928_G_A_b38   G   A  .  . NaN\n",
       "...    ...        ...                     ...  ..  .. .. ..  ..\n",
       "1891  chrX  155528816  chrX_155528816_T_C_b38   C   T  .  . NaN\n",
       "1892  chrX  155649454  chrX_155649454_A_G_b38   G   A  .  . NaN\n",
       "1893  chrX  155668742  chrX_155668742_C_A_b38   A   C  .  . NaN\n",
       "1894  chrX  155679088  chrX_155679088_G_C_b38   G   C  .  . NaN\n",
       "1895  chrX  155687184  chrX_155687184_G_T_b38   G   T  .  . NaN\n",
       "\n",
       "[1896 rows x 8 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_name = '../data/GTEx_eQTL/Enformer_processed/data_gtex_fine_vcf_Whole_Blood_neg.vcf'\n",
    "df = pd.read_csv(file_name, sep='\\t', names=['chr', 'pos', 'variant_id', 'ref', 'alt', '1', '2', '3'], comment='#')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 42%|████▏     | 801/1896 [00:00<00:00, 8008.37it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1896/1896 [00:00<00:00, 3929.77it/s]\n",
      "100%|██████████| 1896/1896 [00:00<00:00, 6044.10it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>seq</th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele</th>\n",
       "      <th>casual_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_63671_G_A_b38</td>\n",
       "      <td>GGGCACCTTCTTTTTCTTAATTGTATCATACATTTTTATTCTGGTC...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63671</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ref</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_63671_G_A_b38</td>\n",
       "      <td>GGGCACCTTCTTTTTCTTAATTGTATCATACATTTTTATTCTGGTC...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>63671</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>alt</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_108826_G_C_b38</td>\n",
       "      <td>TGGATCCTTAATAAACAAGGATGTTTCTGCATCATTTCCCCACAAC...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>108826</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>ref</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_108826_G_C_b38</td>\n",
       "      <td>TGGATCCTTAATAAACAAGGATGTTTCTGCATCATTTCCCCACAAC...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>108826</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>alt</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_135203_G_A_b38</td>\n",
       "      <td>GGAGCTTTGGACTGGGGAGGCCGCAGTGAGGCGAGAGCTAGCTGGG...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>135203</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ref</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7579</th>\n",
       "      <td>chrX_155668742_C_A_b38</td>\n",
       "      <td>TTTGAGCCTGCTTTAAACCATGAAATTCAACACCATCCTCAGATTC...</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155668742</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>alt</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7580</th>\n",
       "      <td>chrX_155679088_G_C_b38</td>\n",
       "      <td>TTGAACAATGAGATCACTTGGACACAGAGCGGGGAACATCACACAC...</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155679088</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>ref</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7581</th>\n",
       "      <td>chrX_155679088_G_C_b38</td>\n",
       "      <td>TTGAACAATGAGATCACTTGGACACAGAGCGGGGAACATCACACAC...</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155679088</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>alt</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7582</th>\n",
       "      <td>chrX_155687184_G_T_b38</td>\n",
       "      <td>AAATAGCTGTTCATGCATTCTGTCCACATTTTGAAGTTCAGTTCAG...</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155687184</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ref</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7583</th>\n",
       "      <td>chrX_155687184_G_T_b38</td>\n",
       "      <td>AAATAGCTGTTCATGCATTCTGTCCACATTTTGAAGTTCAGTTCAG...</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155687184</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>alt</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7584 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  variant_id                                                seq chrom        pos ref alt allele  casual_label\n",
       "0         chr1_63671_G_A_b38  GGGCACCTTCTTTTTCTTAATTGTATCATACATTTTTATTCTGGTC...  chr1      63671   G   A    ref             1\n",
       "1         chr1_63671_G_A_b38  GGGCACCTTCTTTTTCTTAATTGTATCATACATTTTTATTCTGGTC...  chr1      63671   G   A    alt             1\n",
       "2        chr1_108826_G_C_b38  TGGATCCTTAATAAACAAGGATGTTTCTGCATCATTTCCCCACAAC...  chr1     108826   G   C    ref             1\n",
       "3        chr1_108826_G_C_b38  TGGATCCTTAATAAACAAGGATGTTTCTGCATCATTTCCCCACAAC...  chr1     108826   G   C    alt             1\n",
       "4        chr1_135203_G_A_b38  GGAGCTTTGGACTGGGGAGGCCGCAGTGAGGCGAGAGCTAGCTGGG...  chr1     135203   G   A    ref             1\n",
       "...                      ...                                                ...   ...        ...  ..  ..    ...           ...\n",
       "7579  chrX_155668742_C_A_b38  TTTGAGCCTGCTTTAAACCATGAAATTCAACACCATCCTCAGATTC...  chrX  155668742   C   A    alt             0\n",
       "7580  chrX_155679088_G_C_b38  TTGAACAATGAGATCACTTGGACACAGAGCGGGGAACATCACACAC...  chrX  155679088   G   C    ref             0\n",
       "7581  chrX_155679088_G_C_b38  TTGAACAATGAGATCACTTGGACACAGAGCGGGGAACATCACACAC...  chrX  155679088   G   C    alt             0\n",
       "7582  chrX_155687184_G_T_b38  AAATAGCTGTTCATGCATTCTGTCCACATTTTGAAGTTCAGTTCAG...  chrX  155687184   G   T    ref             0\n",
       "7583  chrX_155687184_G_T_b38  AAATAGCTGTTCATGCATTCTGTCCACATTTTGAAGTTCAGTTCAG...  chrX  155687184   G   T    alt             0\n",
       "\n",
       "[7584 rows x 8 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pyfaidx import Fasta\n",
    "import pandas as pd\n",
    "\n",
    "\n",
    "file_name = '../data/GTEx_eQTL/Enformer_processed/data_gtex_fine_vcf_Whole_Blood_pos.vcf'\n",
    "df = pd.read_csv(file_name, sep='\\t', names=['chr', 'pos', 'variant_id', 'ref', 'alt', '1', '2', '3'], comment='#')\n",
    "\n",
    "\n",
    "# 加载参考基因组\n",
    "fasta = Fasta('../../../genome/hg38.fa')\n",
    "\n",
    "\n",
    "seqs = []\n",
    "for _, row in tqdm(df.iterrows(), total=len(df)):\n",
    "    # 解析变异ID: chr1_123456_G_A_b37\n",
    "    variant_id = row['variant_id']\n",
    "    chrom, pos, ref, alt, _ = variant_id.split('_')\n",
    "    pos = int(pos)\n",
    "    \n",
    "    # 提取参考序列\n",
    "    start = pos - 100 # pyfaidx 索引从0开始\n",
    "    end = pos + 99\n",
    "    \n",
    "    try:\n",
    "        seq = fasta[chrom][start:end].seq.upper()\n",
    "        seqs.append({\n",
    "            'variant_id': variant_id,\n",
    "            'seq': seq,\n",
    "            'chrom': chrom,\n",
    "            'pos': pos,\n",
    "            'ref': ref,\n",
    "            'alt': alt,\n",
    "            'allele': 'ref',\n",
    "            'casual_label': 1,\n",
    "        })\n",
    "        assert seq[99] == ref\n",
    "\n",
    "        alt_seq = seq[:99] + alt + seq[100:]\n",
    "        seqs.append({\n",
    "            'variant_id': variant_id,\n",
    "            'seq': alt_seq,\n",
    "            'chrom': chrom,\n",
    "            'pos': pos,\n",
    "            'ref': ref,\n",
    "            'alt': alt,\n",
    "            'allele': 'alt',\n",
    "            'casual_label': 1,\n",
    "        })\n",
    "\n",
    "\n",
    "    except KeyError:\n",
    "        print(f\"Chromosome {chrom} not found in reference.\")\n",
    "        continue\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "file_name = '../data/GTEx_eQTL/Enformer_processed/data_gtex_fine_vcf_Whole_Blood_neg.vcf'\n",
    "df = pd.read_csv(file_name, sep='\\t', names=['chr', 'pos', 'variant_id', 'ref', 'alt', '1', '2', '3'], comment='#')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "for _, row in tqdm(df.iterrows(), total=len(df)):\n",
    "    # 解析变异ID: chr1_123456_G_A_b37\n",
    "    variant_id = row['variant_id']\n",
    "    chrom, pos, ref, alt, _ = variant_id.split('_')\n",
    "    pos = int(pos)\n",
    "    \n",
    "    # 提取参考序列\n",
    "    start = pos - 100 # pyfaidx 索引从0开始\n",
    "    end = pos + 99\n",
    "    \n",
    "    try:\n",
    "        seq = fasta[chrom][start:end].seq.upper()\n",
    "        seqs.append({\n",
    "            'variant_id': variant_id,\n",
    "            'seq': seq,\n",
    "            'chrom': chrom,\n",
    "            'pos': pos,\n",
    "            'ref': ref,\n",
    "            'alt': alt,\n",
    "            'allele': 'ref',\n",
    "            'casual_label': 0,\n",
    "        })\n",
    "        assert seq[99] == ref\n",
    "\n",
    "        alt_seq = seq[:99] + alt + seq[100:]\n",
    "        seqs.append({\n",
    "            'variant_id': variant_id,\n",
    "            'seq': alt_seq,\n",
    "            'chrom': chrom,\n",
    "            'pos': pos,\n",
    "            'ref': ref,\n",
    "            'alt': alt,\n",
    "            'allele': 'alt',\n",
    "            'casual_label': 0,\n",
    "        })\n",
    "\n",
    "\n",
    "    except KeyError:\n",
    "        print(f\"Chromosome {chrom} not found in reference.\")\n",
    "        continue\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "seq_df = pd.DataFrame(seqs)\n",
    "seq_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_df.to_csv('../data/GTEx_eQTL/Enformer_processed/Whole_Blood_data.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "torch",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
